// @angular/material/stepper/typings/index.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from '@angular/material/stepper';
import * as i2 from '@angular/cdk/bidi';
import * as i3 from '@angular/platform-browser';
import * as i4 from '@angular/common';
import * as i5 from '@angular/cdk/platform';
import * as i6 from '@angular/cdk/scrolling';
import * as i7 from '@angular/cdk/a11y';
import * as i8 from '@angular/material/icon';
import * as i9 from '@angular/http';
import * as i10 from '@angular/material/core';
import * as i11 from '@angular/cdk/portal';
import * as i12 from '@angular/material/button';
import * as i13 from '@angular/cdk/stepper';
import * as i14 from '../../icon/typings/index.ngfactory';
export const MdStepperModuleNgFactory:i0.NgModuleFactory<i1.MdStepperModule> = i0.ɵcmf(i1.MdStepperModule,
    ([] as any[]),(_l:any) => {
      return i0.ɵmod([i0.ɵmpd(512,i0.ComponentFactoryResolver,i0.ɵCodegenComponentFactoryResolver,
          [[8,([] as any[])],[3,i0.ComponentFactoryResolver],i0.NgModuleRef]),i0.ɵmpd(6144,
          i2.DIR_DOCUMENT,(null as any),[i3.DOCUMENT]),i0.ɵmpd(4608,i2.Directionality,
          i2.Directionality,[[2,i2.DIR_DOCUMENT]]),i0.ɵmpd(4608,i4.NgLocalization,
          i4.NgLocaleLocalization,[i0.LOCALE_ID]),i0.ɵmpd(4608,i5.Platform,i5.Platform,
          ([] as any[])),i0.ɵmpd(5120,i6.ScrollDispatcher,i6.SCROLL_DISPATCHER_PROVIDER_FACTORY,
          [[3,i6.ScrollDispatcher],i0.NgZone,i5.Platform]),i0.ɵmpd(5120,i6.ViewportRuler,
          i6.VIEWPORT_RULER_PROVIDER_FACTORY,[[3,i6.ViewportRuler],i6.ScrollDispatcher]),
          i0.ɵmpd(4608,i7.InteractivityChecker,i7.InteractivityChecker,[i5.Platform]),
          i0.ɵmpd(4608,i7.FocusTrapFactory,i7.FocusTrapFactory,[i7.InteractivityChecker,
              i5.Platform,i0.NgZone]),i0.ɵmpd(136192,i7.AriaDescriber,i7.ARIA_DESCRIBER_PROVIDER_FACTORY,
              [[3,i7.AriaDescriber],i5.Platform]),i0.ɵmpd(5120,i7.LiveAnnouncer,i7.LIVE_ANNOUNCER_PROVIDER_FACTORY,
              [[3,i7.LiveAnnouncer],[2,i7.LIVE_ANNOUNCER_ELEMENT_TOKEN],i5.Platform]),
          i0.ɵmpd(5120,i7.FocusMonitor,i7.FOCUS_MONITOR_PROVIDER_FACTORY,[[3,i7.FocusMonitor],
              i0.NgZone,i5.Platform]),i0.ɵmpd(5120,i8.MdIconRegistry,i8.ICON_REGISTRY_PROVIDER_FACTORY,
              [[3,i8.MdIconRegistry],[2,i9.Http],i3.DomSanitizer]),i0.ɵmpd(512,i10.CompatibilityModule,
              i10.CompatibilityModule,([] as any[])),i0.ɵmpd(512,i2.BidiModule,i2.BidiModule,
              ([] as any[])),i0.ɵmpd(256,i10.MATERIAL_SANITY_CHECKS,true,([] as any[])),
          i0.ɵmpd(512,i10.MdCommonModule,i10.MdCommonModule,[[2,i3.DOCUMENT],[2,i10.MATERIAL_SANITY_CHECKS]]),
          i0.ɵmpd(512,i4.CommonModule,i4.CommonModule,([] as any[])),i0.ɵmpd(512,i11.PortalModule,
              i11.PortalModule,([] as any[])),i0.ɵmpd(512,i5.PlatformModule,i5.PlatformModule,
              ([] as any[])),i0.ɵmpd(512,i6.ScrollDispatchModule,i6.ScrollDispatchModule,
              ([] as any[])),i0.ɵmpd(512,i10.MdRippleModule,i10.MdRippleModule,([] as any[])),
          i0.ɵmpd(512,i7.A11yModule,i7.A11yModule,([] as any[])),i0.ɵmpd(512,i12.MdButtonModule,
              i12.MdButtonModule,([] as any[])),i0.ɵmpd(512,i13.CdkStepperModule,i13.CdkStepperModule,
              ([] as any[])),i0.ɵmpd(512,i8.MdIconModule,i8.MdIconModule,([] as any[])),
          i0.ɵmpd(512,i1.MdStepperModule,i1.MdStepperModule,([] as any[]))]);
    });
const styles_MdHorizontalStepper:any[] = ['.mat-stepper-horizontal,.mat-stepper-vertical{display:block}.mat-step-header{overflow:hidden;outline:0}.mat-horizontal-stepper-header-container{white-space:nowrap;display:flex;align-items:center}.mat-stepper-horizontal-line{border-top-width:1px;border-top-style:solid;flex:auto;height:0;margin:0 -16px;min-width:32px}.mat-horizontal-stepper-header{display:flex;height:72px;overflow:hidden;align-items:center;padding:0 24px}.mat-horizontal-stepper-header .mat-step-icon,.mat-horizontal-stepper-header .mat-step-icon-not-touched{margin-right:8px;flex:none}[dir=rtl] .mat-horizontal-stepper-header .mat-step-icon,[dir=rtl] .mat-horizontal-stepper-header .mat-step-icon-not-touched{margin-right:0;margin-left:8px}.mat-vertical-stepper-header{display:flex;align-items:center;padding:24px;max-height:24px}.mat-vertical-stepper-header .mat-step-icon,.mat-vertical-stepper-header .mat-step-icon-not-touched{margin-right:12px}[dir=rtl] .mat-vertical-stepper-header .mat-step-icon,[dir=rtl] .mat-vertical-stepper-header .mat-step-icon-not-touched{margin-right:0;margin-left:12px}.mat-horizontal-stepper-content{overflow:hidden}.mat-horizontal-stepper-content[aria-expanded=false]{height:0}.mat-horizontal-content-container{overflow:hidden;padding:0 24px 24px 24px}.mat-vertical-content-container{margin-left:36px;border:0;position:relative}[dir=rtl] .mat-vertical-content-container{margin-left:0;margin-right:36px}.mat-stepper-vertical-line::before{content:\'\';position:absolute;top:-16px;bottom:-16px;left:0;border-left-width:1px;border-left-style:solid}[dir=rtl] .mat-stepper-vertical-line::before{left:auto;right:0}.mat-vertical-stepper-content{overflow:hidden}.mat-vertical-content{padding:0 24px 24px 24px}.mat-step:last-child .mat-vertical-content-container{border:none}'];
export const RenderType_MdHorizontalStepper:i0.RendererType2 = i0.ɵcrt({encapsulation:2,
    styles:styles_MdHorizontalStepper,data:{'animation':[{type:7,name:'stepTransition',
        definitions:[{type:0,name:'previous',styles:{type:6,styles:{transform:'translate3d(-100%, 0, 0)',
            visibility:'hidden'},offset:(null as any)},options:(undefined as any)},
            {type:0,name:'current',styles:{type:6,styles:{transform:'translate3d(0%, 0, 0)',
                visibility:'visible'},offset:(null as any)},options:(undefined as any)},
            {type:0,name:'next',styles:{type:6,styles:{transform:'translate3d(100%, 0, 0)',
                visibility:'hidden'},offset:(null as any)},options:(undefined as any)},
            {type:1,expr:'* => *',animation:{type:4,styles:(null as any),timings:'500ms cubic-bezier(0.35, 0, 0.25, 1)'},
                options:(null as any)}],options:{}}]}});
function View_MdHorizontalStepper_2(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),0,'div',[['class',
      'mat-stepper-horizontal-line']],(null as any),(null as any),(null as any),(null as any),
      (null as any)))],(null as any),(null as any));
}
function View_MdHorizontalStepper_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),5,(null as any),
      (null as any),(null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵeld(0,[[1,0]],(null as any),2,'mat-step-header',[['class','mat-horizontal-stepper-header mat-step-header'],
          ['role','tab']],[[8,'tabIndex',0],[8,'id',0],[1,'aria-controls',0],[1,'aria-selected',
          0]],[[(null as any),'click'],[(null as any),'keydown']],(_v,en,$event) => {
        var ad:boolean = true;
        var _co:any = _v.component;
        if (('click' === en)) {
          const pd_0:any = ((<any>_v.context.$implicit.select()) !== false);
          ad = (pd_0 && ad);
        }
        if (('keydown' === en)) {
          const pd_1:any = ((<any>_co._onKeydown($event)) !== false);
          ad = (pd_1 && ad);
        }
        return ad;
      },View_MdStepHeader_0,RenderType_MdStepHeader)),i0.ɵdid(49152,(null as any),
          0,i1.MdStepHeader,([] as any[]),{icon:[0,'icon'],label:[1,'label'],index:[2,
              'index'],selected:[3,'selected'],active:[4,'active'],optional:[5,'optional']},
          (null as any)),i0.ɵprd(8448,(null as any),i10.MATERIAL_COMPATIBILITY_MODE,
          true,([] as any[])),(_l()(),i0.ɵand(16777216,(null as any),(null as any),
          1,(null as any),View_MdHorizontalStepper_2)),i0.ɵdid(16384,(null as any),
          0,i4.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),
      (_l()(),i0.ɵand(0,(null as any),(null as any),0))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_4:any = _co._getIndicatorType(_v.context.index);
    const currVal_5:any = (_v.context.$implicit.stepLabel || _v.context.$implicit.label);
    const currVal_6:any = _v.context.index;
    const currVal_7:any = (_co.selectedIndex === _v.context.index);
    const currVal_8:any = (_v.context.$implicit.completed || (_co.selectedIndex === _v.context.index));
    const currVal_9:any = _v.context.$implicit.optional;
    _ck(_v,2,0,currVal_4,currVal_5,currVal_6,currVal_7,currVal_8,currVal_9);
    const currVal_10:boolean = !_v.context.last;
    _ck(_v,5,0,currVal_10);
  },(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = ((_co._focusIndex === _v.context.index)? 0: (0 - 1));
    const currVal_1:any = _co._getStepLabelId(_v.context.index);
    const currVal_2:any = _co._getStepContentId(_v.context.index);
    const currVal_3:any = (_co.selectedIndex == _v.context.index);
    _ck(_v,1,0,currVal_0,currVal_1,currVal_2,currVal_3);
  });
}
function View_MdHorizontalStepper_3(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),2,'div',[['class',
      'mat-horizontal-stepper-content'],['role','tabpanel']],[[24,'@stepTransition',
      0],[8,'id',0],[1,'aria-labelledby',0],[1,'aria-expanded',0]],(null as any),(null as any),
      (null as any),(null as any))),(_l()(),i0.ɵeld(16777216,(null as any),(null as any),
      1,(null as any),(null as any),(null as any),(null as any),(null as any),(null as any),
      (null as any))),i0.ɵdid(540672,(null as any),0,i4.NgTemplateOutlet,[i0.ViewContainerRef],
      {ngTemplateOutlet:[0,'ngTemplateOutlet']},(null as any))],(_ck,_v) => {
    const currVal_4:any = _v.context.$implicit.content;
    _ck(_v,2,0,currVal_4);
  },(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co._getAnimationDirection(_v.context.index);
    const currVal_1:any = _co._getStepContentId(_v.context.index);
    const currVal_2:any = _co._getStepLabelId(_v.context.index);
    const currVal_3:any = (_co.selectedIndex === _v.context.index);
    _ck(_v,0,0,currVal_0,currVal_1,currVal_2,currVal_3);
  });
}
export function View_MdHorizontalStepper_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[i0.ɵqud(671088640,1,{_stepHeader:1}),(_l()(),i0.ɵeld(0,(null as any),
      (null as any),2,'div',[['class','mat-horizontal-stepper-header-container']],
      (null as any),(null as any),(null as any),(null as any),(null as any))),(_l()(),
      i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_MdHorizontalStepper_1)),
      i0.ɵdid(802816,(null as any),0,i4.NgForOf,[i0.ViewContainerRef,i0.TemplateRef,
          i0.IterableDiffers],{ngForOf:[0,'ngForOf']},(null as any)),(_l()(),i0.ɵeld(0,
          (null as any),(null as any),2,'div',[['class','mat-horizontal-content-container']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_MdHorizontalStepper_3)),
      i0.ɵdid(802816,(null as any),0,i4.NgForOf,[i0.ViewContainerRef,i0.TemplateRef,
          i0.IterableDiffers],{ngForOf:[0,'ngForOf']},(null as any))],(_ck,_v) => {
    var _co:i1.MdHorizontalStepper = _v.component;
    const currVal_0:any = _co._steps;
    _ck(_v,3,0,currVal_0);
    const currVal_1:any = _co._steps;
    _ck(_v,6,0,currVal_1);
  },(null as any));
}
export function View_MdHorizontalStepper_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),4,'md-horizontal-stepper',
      [['class','mat-stepper-horizontal'],['role','tablist']],(null as any),(null as any),
      (null as any),View_MdHorizontalStepper_0,RenderType_MdHorizontalStepper)),i0.ɵprd(6144,
      (null as any),i1.MdStepper,(null as any),[i1.MdHorizontalStepper]),i0.ɵdid(49152,
      (null as any),1,i1.MdHorizontalStepper,[[2,i2.Directionality]],(null as any),
      (null as any)),i0.ɵqud(603979776,1,{_steps:1}),i0.ɵprd(8448,(null as any),i10.MATERIAL_COMPATIBILITY_MODE,
      true,([] as any[]))],(null as any),(null as any));
}
export const MdHorizontalStepperNgFactory:i0.ComponentFactory<i1.MdHorizontalStepper> = i0.ɵccf('md-horizontal-stepper, mat-horizontal-stepper',
    i1.MdHorizontalStepper,View_MdHorizontalStepper_Host_0,{linear:'linear',selectedIndex:'selectedIndex',
        selected:'selected'},{selectionChange:'selectionChange'},([] as any[]));
const styles_MdVerticalStepper:any[] = ['.mat-stepper-horizontal,.mat-stepper-vertical{display:block}.mat-step-header{overflow:hidden;outline:0}.mat-horizontal-stepper-header-container{white-space:nowrap;display:flex;align-items:center}.mat-stepper-horizontal-line{border-top-width:1px;border-top-style:solid;flex:auto;height:0;margin:0 -16px;min-width:32px}.mat-horizontal-stepper-header{display:flex;height:72px;overflow:hidden;align-items:center;padding:0 24px}.mat-horizontal-stepper-header .mat-step-icon,.mat-horizontal-stepper-header .mat-step-icon-not-touched{margin-right:8px;flex:none}[dir=rtl] .mat-horizontal-stepper-header .mat-step-icon,[dir=rtl] .mat-horizontal-stepper-header .mat-step-icon-not-touched{margin-right:0;margin-left:8px}.mat-vertical-stepper-header{display:flex;align-items:center;padding:24px;max-height:24px}.mat-vertical-stepper-header .mat-step-icon,.mat-vertical-stepper-header .mat-step-icon-not-touched{margin-right:12px}[dir=rtl] .mat-vertical-stepper-header .mat-step-icon,[dir=rtl] .mat-vertical-stepper-header .mat-step-icon-not-touched{margin-right:0;margin-left:12px}.mat-horizontal-stepper-content{overflow:hidden}.mat-horizontal-stepper-content[aria-expanded=false]{height:0}.mat-horizontal-content-container{overflow:hidden;padding:0 24px 24px 24px}.mat-vertical-content-container{margin-left:36px;border:0;position:relative}[dir=rtl] .mat-vertical-content-container{margin-left:0;margin-right:36px}.mat-stepper-vertical-line::before{content:\'\';position:absolute;top:-16px;bottom:-16px;left:0;border-left-width:1px;border-left-style:solid}[dir=rtl] .mat-stepper-vertical-line::before{left:auto;right:0}.mat-vertical-stepper-content{overflow:hidden}.mat-vertical-content{padding:0 24px 24px 24px}.mat-step:last-child .mat-vertical-content-container{border:none}'];
export const RenderType_MdVerticalStepper:i0.RendererType2 = i0.ɵcrt({encapsulation:2,
    styles:styles_MdVerticalStepper,data:{'animation':[{type:7,name:'stepTransition',
        definitions:[{type:0,name:'previous',styles:{type:6,styles:{height:'0px',visibility:'hidden'},
            offset:(null as any)},options:(undefined as any)},{type:0,name:'next',
            styles:{type:6,styles:{height:'0px',visibility:'hidden'},offset:(null as any)},
            options:(undefined as any)},{type:0,name:'current',styles:{type:6,styles:{height:'*',
            visibility:'visible'},offset:(null as any)},options:(undefined as any)},
            {type:1,expr:'* <=> current',animation:{type:4,styles:(null as any),timings:'225ms cubic-bezier(0.4, 0.0, 0.2, 1)'},
                options:(null as any)}],options:{}}]}});
function View_MdVerticalStepper_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),8,'div',[['class',
      'mat-step']],(null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵeld(0,[[1,0]],(null as any),2,'mat-step-header',[['class','mat-vertical-stepper-header mat-step-header'],
          ['role','tab']],[[8,'tabIndex',0],[8,'id',0],[1,'aria-controls',0],[1,'aria-selected',
          0]],[[(null as any),'click'],[(null as any),'keydown']],(_v,en,$event) => {
        var ad:boolean = true;
        var _co:any = _v.component;
        if (('click' === en)) {
          const pd_0:any = ((<any>_v.context.$implicit.select()) !== false);
          ad = (pd_0 && ad);
        }
        if (('keydown' === en)) {
          const pd_1:any = ((<any>_co._onKeydown($event)) !== false);
          ad = (pd_1 && ad);
        }
        return ad;
      },View_MdStepHeader_0,RenderType_MdStepHeader)),i0.ɵdid(49152,(null as any),
          0,i1.MdStepHeader,([] as any[]),{icon:[0,'icon'],label:[1,'label'],index:[2,
              'index'],selected:[3,'selected'],active:[4,'active'],optional:[5,'optional']},
          (null as any)),i0.ɵprd(8448,(null as any),i10.MATERIAL_COMPATIBILITY_MODE,
          true,([] as any[])),(_l()(),i0.ɵeld(0,(null as any),(null as any),4,'div',
          [['class','mat-vertical-content-container']],[[2,'mat-stepper-vertical-line',
              (null as any)]],(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),3,'div',[['class','mat-vertical-stepper-content'],
          ['role','tabpanel']],[[24,'@stepTransition',0],[8,'id',0],[1,'aria-labelledby',
          0],[1,'aria-expanded',0]],(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),2,'div',[['class','mat-vertical-content']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵeld(16777216,(null as any),(null as any),1,(null as any),(null as any),
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      i0.ɵdid(540672,(null as any),0,i4.NgTemplateOutlet,[i0.ViewContainerRef],{ngTemplateOutlet:[0,
          'ngTemplateOutlet']},(null as any))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_4:any = _co._getIndicatorType(_v.context.index);
    const currVal_5:any = (_v.context.$implicit.stepLabel || _v.context.$implicit.label);
    const currVal_6:any = _v.context.index;
    const currVal_7:any = (_co.selectedIndex === _v.context.index);
    const currVal_8:any = (_v.context.$implicit.completed || (_co.selectedIndex === _v.context.index));
    const currVal_9:any = _v.context.$implicit.optional;
    _ck(_v,2,0,currVal_4,currVal_5,currVal_6,currVal_7,currVal_8,currVal_9);
    const currVal_15:any = _v.context.$implicit.content;
    _ck(_v,8,0,currVal_15);
  },(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = ((_co._focusIndex == _v.context.index)? 0: (0 - 1));
    const currVal_1:any = _co._getStepLabelId(_v.context.index);
    const currVal_2:any = _co._getStepContentId(_v.context.index);
    const currVal_3:any = (_co.selectedIndex === _v.context.index);
    _ck(_v,1,0,currVal_0,currVal_1,currVal_2,currVal_3);
    const currVal_10:boolean = !_v.context.last;
    _ck(_v,4,0,currVal_10);
    const currVal_11:any = _co._getAnimationDirection(_v.context.index);
    const currVal_12:any = _co._getStepContentId(_v.context.index);
    const currVal_13:any = _co._getStepLabelId(_v.context.index);
    const currVal_14:any = (_co.selectedIndex === _v.context.index);
    _ck(_v,5,0,currVal_11,currVal_12,currVal_13,currVal_14);
  });
}
export function View_MdVerticalStepper_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[i0.ɵqud(671088640,1,{_stepHeader:1}),(_l()(),i0.ɵand(16777216,
      (null as any),(null as any),1,(null as any),View_MdVerticalStepper_1)),i0.ɵdid(802816,
      (null as any),0,i4.NgForOf,[i0.ViewContainerRef,i0.TemplateRef,i0.IterableDiffers],
      {ngForOf:[0,'ngForOf']},(null as any))],(_ck,_v) => {
    var _co:i1.MdVerticalStepper = _v.component;
    const currVal_0:any = _co._steps;
    _ck(_v,2,0,currVal_0);
  },(null as any));
}
export function View_MdVerticalStepper_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),4,'md-vertical-stepper',
      [['class','mat-stepper-vertical'],['role','tablist']],(null as any),(null as any),
      (null as any),View_MdVerticalStepper_0,RenderType_MdVerticalStepper)),i0.ɵprd(6144,
      (null as any),i1.MdStepper,(null as any),[i1.MdVerticalStepper]),i0.ɵdid(49152,
      (null as any),1,i1.MdVerticalStepper,[[2,i2.Directionality]],(null as any),(null as any)),
      i0.ɵqud(603979776,1,{_steps:1}),i0.ɵprd(8448,(null as any),i10.MATERIAL_COMPATIBILITY_MODE,
          true,([] as any[]))],(null as any),(null as any));
}
export const MdVerticalStepperNgFactory:i0.ComponentFactory<i1.MdVerticalStepper> = i0.ɵccf('md-vertical-stepper, mat-vertical-stepper',
    i1.MdVerticalStepper,View_MdVerticalStepper_Host_0,{linear:'linear',selectedIndex:'selectedIndex',
        selected:'selected'},{selectionChange:'selectionChange'},([] as any[]));
const styles_MdStep:any[] = ([] as any[]);
export const RenderType_MdStep:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_MdStep,
    data:{}});
function View_MdStep_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[i0.ɵncd((null as any),0),(_l()(),i0.ɵand(0,(null as any),(null as any),
      0))],(null as any),(null as any));
}
export function View_MdStep_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[i0.ɵqud(402653184,1,{content:0}),(_l()(),i0.ɵand(0,[[1,2]],(null as any),
      0,(null as any),View_MdStep_1))],(null as any),(null as any));
}
export function View_MdStep_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),3,'md-step',([] as any[]),
      (null as any),(null as any),(null as any),View_MdStep_0,RenderType_MdStep)),
      i0.ɵprd(6144,(null as any),i10.MD_ERROR_GLOBAL_OPTIONS,(null as any),[i1.MdStep]),
      i0.ɵdid(49152,(null as any),1,i1.MdStep,[i1.MdStepper,[3,i10.MD_ERROR_GLOBAL_OPTIONS]],
          (null as any),(null as any)),i0.ɵqud(335544320,1,{stepLabel:0})],(null as any),
      (null as any));
}
export const MdStepNgFactory:i0.ComponentFactory<i1.MdStep> = i0.ɵccf('md-step, mat-step',
    i1.MdStep,View_MdStep_Host_0,{stepControl:'stepControl',label:'label',editable:'editable',
        optional:'optional',completed:'completed'},{},['*']);
const styles_MdStepHeader:any[] = ['.mat-step-optional{font-size:12px}.mat-step-icon,.mat-step-icon-not-touched{border-radius:50%;height:24px;width:24px;align-items:center;justify-content:center;display:flex}.mat-step-icon .mat-icon{font-size:16px;height:16px;width:16px}.mat-step-label{display:inline-block;white-space:nowrap;overflow:hidden;text-overflow:ellipsis;min-width:50px;vertical-align:middle}.mat-step-text-label{text-overflow:ellipsis;overflow:hidden}'];
export const RenderType_MdStepHeader:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_MdStepHeader,
    data:{}});
function View_MdStepHeader_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'span',([] as any[]),
      (null as any),(null as any),(null as any),(null as any),(null as any))),(_l()(),
      i0.ɵted((null as any),['','']))],(null as any),(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = (_co.index + 1);
    _ck(_v,1,0,currVal_0);
  });
}
function View_MdStepHeader_2(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),3,'mat-icon',[['class',
      'mat-icon'],['role','img']],(null as any),(null as any),(null as any),i14.View_MdIcon_0,
      i14.RenderType_MdIcon)),i0.ɵdid(16384,(null as any),0,i10.MatPrefixRejector,
      [[2,i10.MATERIAL_COMPATIBILITY_MODE],i0.ElementRef],(null as any),(null as any)),
      i0.ɵdid(638976,(null as any),0,i8.MdIcon,[i0.Renderer2,i0.ElementRef,i8.MdIconRegistry,
          [8,(null as any)]],(null as any),(null as any)),(_l()(),i0.ɵted(0,['create']))],
      (_ck,_v) => {
        _ck(_v,2,0);
      },(null as any));
}
function View_MdStepHeader_3(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),3,'mat-icon',[['class',
      'mat-icon'],['role','img']],(null as any),(null as any),(null as any),i14.View_MdIcon_0,
      i14.RenderType_MdIcon)),i0.ɵdid(16384,(null as any),0,i10.MatPrefixRejector,
      [[2,i10.MATERIAL_COMPATIBILITY_MODE],i0.ElementRef],(null as any),(null as any)),
      i0.ɵdid(638976,(null as any),0,i8.MdIcon,[i0.Renderer2,i0.ElementRef,i8.MdIconRegistry,
          [8,(null as any)]],(null as any),(null as any)),(_l()(),i0.ɵted(0,['done']))],
      (_ck,_v) => {
        _ck(_v,2,0);
      },(null as any));
}
function View_MdStepHeader_4(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(16777216,(null as any),(null as any),1,(null as any),
      (null as any),(null as any),(null as any),(null as any),(null as any),(null as any))),
      i0.ɵdid(540672,(null as any),0,i4.NgTemplateOutlet,[i0.ViewContainerRef],{ngTemplateOutlet:[0,
          'ngTemplateOutlet']},(null as any)),(_l()(),i0.ɵand(0,(null as any),(null as any),
          0))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.label.template;
    _ck(_v,1,0,currVal_0);
  },(null as any));
}
function View_MdStepHeader_5(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'div',[['class',
      'mat-step-text-label']],(null as any),(null as any),(null as any),(null as any),
      (null as any))),(_l()(),i0.ɵted((null as any),['','']))],(null as any),(_ck,
      _v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.label;
    _ck(_v,1,0,currVal_0);
  });
}
function View_MdStepHeader_6(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'div',[['class',
      'mat-step-optional']],(null as any),(null as any),(null as any),(null as any),
      (null as any))),(_l()(),i0.ɵted((null as any),['Optional']))],(null as any),
      (null as any));
}
export function View_MdStepHeader_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),7,'div',([] as any[]),
      [[2,'mat-step-icon',(null as any)],[2,'mat-step-icon-not-touched',(null as any)]],
      (null as any),(null as any),(null as any),(null as any))),i0.ɵdid(16384,(null as any),
      0,i4.NgSwitch,([] as any[]),{ngSwitch:[0,'ngSwitch']},(null as any)),(_l()(),
      i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_MdStepHeader_1)),
      i0.ɵdid(278528,(null as any),0,i4.NgSwitchCase,[i0.ViewContainerRef,i0.TemplateRef,
          i4.NgSwitch],{ngSwitchCase:[0,'ngSwitchCase']},(null as any)),(_l()(),i0.ɵand(16777216,
          (null as any),(null as any),1,(null as any),View_MdStepHeader_2)),i0.ɵdid(278528,
          (null as any),0,i4.NgSwitchCase,[i0.ViewContainerRef,i0.TemplateRef,i4.NgSwitch],
          {ngSwitchCase:[0,'ngSwitchCase']},(null as any)),(_l()(),i0.ɵand(16777216,
          (null as any),(null as any),1,(null as any),View_MdStepHeader_3)),i0.ɵdid(278528,
          (null as any),0,i4.NgSwitchCase,[i0.ViewContainerRef,i0.TemplateRef,i4.NgSwitch],
          {ngSwitchCase:[0,'ngSwitchCase']},(null as any)),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),6,'div',[['class','mat-step-label']],[[2,'mat-step-label-active',
              (null as any)]],(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_MdStepHeader_4)),
      i0.ɵdid(16384,(null as any),0,i4.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵand(16777216,(null as any),(null as any),
          1,(null as any),View_MdStepHeader_5)),i0.ɵdid(16384,(null as any),0,i4.NgIf,
          [i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),(_l()(),
          i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_MdStepHeader_6)),
      i0.ɵdid(16384,(null as any),0,i4.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any))],(_ck,_v) => {
    var _co:i1.MdStepHeader = _v.component;
    const currVal_2:any = _co.icon;
    _ck(_v,1,0,currVal_2);
    const currVal_3:any = 'number';
    _ck(_v,3,0,currVal_3);
    const currVal_4:any = 'edit';
    _ck(_v,5,0,currVal_4);
    const currVal_5:any = 'done';
    _ck(_v,7,0,currVal_5);
    const currVal_7:any = _co._templateLabel();
    _ck(_v,10,0,currVal_7);
    const currVal_8:any = _co._stringLabel();
    _ck(_v,12,0,currVal_8);
    const currVal_9:any = _co.optional;
    _ck(_v,14,0,currVal_9);
  },(_ck,_v) => {
    var _co:i1.MdStepHeader = _v.component;
    const currVal_0:any = ((_co.icon !== 'number') || _co.selected);
    const currVal_1:any = ((_co.icon == 'number') && !_co.selected);
    _ck(_v,0,0,currVal_0,currVal_1);
    const currVal_6:any = _co.active;
    _ck(_v,8,0,currVal_6);
  });
}
export function View_MdStepHeader_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),2,'md-step-header',
      [['class','mat-step-header'],['role','tab']],(null as any),(null as any),(null as any),
      View_MdStepHeader_0,RenderType_MdStepHeader)),i0.ɵdid(49152,(null as any),0,
      i1.MdStepHeader,([] as any[]),(null as any),(null as any)),i0.ɵprd(8448,(null as any),
      i10.MATERIAL_COMPATIBILITY_MODE,true,([] as any[]))],(null as any),(null as any));
}
export const MdStepHeaderNgFactory:i0.ComponentFactory<i1.MdStepHeader> = i0.ɵccf('md-step-header, mat-step-header',
    i1.MdStepHeader,View_MdStepHeader_Host_0,{icon:'icon',label:'label',index:'index',
        selected:'selected',active:'active',optional:'optional'},{},([] as any[]));
