// @angular/material/expansion/typings/index.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from '@angular/material/expansion';
import * as i2 from '@angular/common';
import * as i3 from '@angular/cdk/platform';
import * as i4 from '@angular/cdk/a11y';
import * as i5 from '@angular/material/core';
export const MdExpansionModuleNgFactory:i0.NgModuleFactory<i1.MdExpansionModule> = i0.ɵcmf(i1.MdExpansionModule,
    ([] as any[]),(_l:any) => {
      return i0.ɵmod([i0.ɵmpd(512,i0.ComponentFactoryResolver,i0.ɵCodegenComponentFactoryResolver,
          [[8,([] as any[])],[3,i0.ComponentFactoryResolver],i0.NgModuleRef]),i0.ɵmpd(4608,
          i2.NgLocalization,i2.NgLocaleLocalization,[i0.LOCALE_ID]),i0.ɵmpd(4608,i3.Platform,
          i3.Platform,([] as any[])),i0.ɵmpd(4608,i4.InteractivityChecker,i4.InteractivityChecker,
          [i3.Platform]),i0.ɵmpd(4608,i4.FocusTrapFactory,i4.FocusTrapFactory,[i4.InteractivityChecker,
          i3.Platform,i0.NgZone]),i0.ɵmpd(136192,i4.AriaDescriber,i4.ARIA_DESCRIBER_PROVIDER_FACTORY,
          [[3,i4.AriaDescriber],i3.Platform]),i0.ɵmpd(5120,i4.LiveAnnouncer,i4.LIVE_ANNOUNCER_PROVIDER_FACTORY,
          [[3,i4.LiveAnnouncer],[2,i4.LIVE_ANNOUNCER_ELEMENT_TOKEN],i3.Platform]),
          i0.ɵmpd(5120,i4.FocusMonitor,i4.FOCUS_MONITOR_PROVIDER_FACTORY,[[3,i4.FocusMonitor],
              i0.NgZone,i3.Platform]),i0.ɵmpd(5120,i5.UniqueSelectionDispatcher,i5.UNIQUE_SELECTION_DISPATCHER_PROVIDER_FACTORY,
              [[3,i5.UniqueSelectionDispatcher]]),i0.ɵmpd(512,i5.CompatibilityModule,
              i5.CompatibilityModule,([] as any[])),i0.ɵmpd(512,i2.CommonModule,i2.CommonModule,
              ([] as any[])),i0.ɵmpd(512,i3.PlatformModule,i3.PlatformModule,([] as any[])),
          i0.ɵmpd(512,i4.A11yModule,i4.A11yModule,([] as any[])),i0.ɵmpd(512,i1.MdExpansionModule,
              i1.MdExpansionModule,([] as any[]))]);
    });
const styles_MdExpansionPanel:any[] = ['.mat-expansion-panel{transition:box-shadow 280ms cubic-bezier(.4,0,.2,1);box-sizing:content-box;display:block;margin:0;transition:margin 225ms cubic-bezier(.4,0,.2,1)}.mat-expansion-panel:not([class*=mat-elevation-z]){box-shadow:0 3px 1px -2px rgba(0,0,0,.2),0 2px 2px 0 rgba(0,0,0,.14),0 1px 5px 0 rgba(0,0,0,.12)}.mat-expansion-panel-content{overflow:hidden}.mat-expansion-panel-body{padding:0 24px 16px}.mat-expansion-panel-spacing{margin:16px 0}.mat-accordion .mat-expansion-panel-spacing:first-child{margin-top:0}.mat-accordion .mat-expansion-panel-spacing:last-child{margin-bottom:0}.mat-action-row{border-top-style:solid;border-top-width:1px;display:flex;flex-direction:row;justify-content:flex-end;padding:16px 8px 16px 24px}.mat-action-row button.mat-button{margin-left:8px}[dir=rtl] .mat-action-row button.mat-button{margin-left:0;margin-right:8px}'];
export const RenderType_MdExpansionPanel:i0.RendererType2 = i0.ɵcrt({encapsulation:2,
    styles:styles_MdExpansionPanel,data:{'animation':[{type:7,name:'bodyExpansion',
        definitions:[{type:0,name:'collapsed',styles:{type:6,styles:{height:'0px',
            visibility:'hidden'},offset:(null as any)},options:(undefined as any)},
            {type:0,name:'expanded',styles:{type:6,styles:{height:'*',visibility:'visible'},
                offset:(null as any)},options:(undefined as any)},{type:1,expr:'expanded <=> collapsed',
                animation:{type:4,styles:(null as any),timings:'225ms cubic-bezier(0.4,0.0,0.2,1)'},
                options:(null as any)}],options:{}}]}});
export function View_MdExpansionPanel_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(2,[i0.ɵncd((null as any),0),(_l()(),i0.ɵeld(0,(null as any),(null as any),
      3,'div',[['class','mat-expansion-panel-content']],[[2,'mat-expanded',(null as any)],
          [24,'@bodyExpansion',0],[8,'id',0]],(null as any),(null as any),(null as any),
      (null as any))),(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'div',[['class',
      'mat-expansion-panel-body']],(null as any),(null as any),(null as any),(null as any),
      (null as any))),i0.ɵncd((null as any),1),i0.ɵncd((null as any),2)],(null as any),
      (_ck,_v) => {
        var _co:i1.MdExpansionPanel = _v.component;
        const currVal_0:any = _co.expanded;
        const currVal_1:any = _co._getExpandedState();
        const currVal_2:any = _co.id;
        _ck(_v,1,0,currVal_0,currVal_1,currVal_2);
      });
}
export function View_MdExpansionPanel_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),2,'md-expansion-panel',
      [['class','mat-expansion-panel']],[[2,'mat-expanded',(null as any)],[2,'mat-expansion-panel-spacing',
          (null as any)]],(null as any),(null as any),View_MdExpansionPanel_0,RenderType_MdExpansionPanel)),
      i0.ɵprd(6144,(null as any),i1.AccordionItem,(null as any),[i1.MdExpansionPanel]),
      i0.ɵdid(704512,(null as any),0,i1.MdExpansionPanel,[[2,i1.MdAccordion],i0.ChangeDetectorRef,
          i5.UniqueSelectionDispatcher],(null as any),(null as any))],(null as any),
      (_ck,_v) => {
        const currVal_0:any = i0.ɵnov(_v,2).expanded;
        const currVal_1:any = i0.ɵnov(_v,2)._hasSpacing();
        _ck(_v,0,0,currVal_0,currVal_1);
      });
}
export const MdExpansionPanelNgFactory:i0.ComponentFactory<i1.MdExpansionPanel> = i0.ɵccf('md-expansion-panel, mat-expansion-panel',
    i1.MdExpansionPanel,View_MdExpansionPanel_Host_0,{disabled:'disabled',expanded:'expanded',
        hideToggle:'hideToggle'},{},['mat-expansion-panel-header, md-expansion-panel-header',
        '*','mat-action-row, md-action-row']);
const styles_MdExpansionPanelHeader:any[] = ['.mat-expansion-panel-header{display:flex;flex-direction:row;align-items:center;padding:0 24px}.mat-expansion-panel-header:focus,.mat-expansion-panel-header:hover{outline:0}.mat-expansion-panel-header.mat-expanded:focus,.mat-expansion-panel-header.mat-expanded:hover{background:inherit}.mat-expansion-panel-header:not([aria-disabled=true]){cursor:pointer}.mat-content{display:flex;flex:1;flex-direction:row;overflow:hidden}.mat-expansion-panel-header-description,.mat-expansion-panel-header-title{display:flex;flex-grow:1;margin-right:16px}[dir=rtl] .mat-expansion-panel-header-description,[dir=rtl] .mat-expansion-panel-header-title{margin-right:0;margin-left:16px}.mat-expansion-panel-header-description{flex-grow:2}.mat-expansion-indicator::after{border-style:solid;border-width:0 2px 2px 0;content:\'\';display:inline-block;padding:3px;transform:rotate(45deg);vertical-align:middle}'];
export const RenderType_MdExpansionPanelHeader:i0.RendererType2 = i0.ɵcrt({encapsulation:2,
    styles:styles_MdExpansionPanelHeader,data:{'animation':[{type:7,name:'indicatorRotate',
        definitions:[{type:0,name:'collapsed',styles:{type:6,styles:{transform:'rotate(0deg)'},
            offset:(null as any)},options:(undefined as any)},{type:0,name:'expanded',
            styles:{type:6,styles:{transform:'rotate(180deg)'},offset:(null as any)},
            options:(undefined as any)},{type:1,expr:'expanded <=> collapsed',animation:{type:4,
            styles:(null as any),timings:'225ms cubic-bezier(0.4,0.0,0.2,1)'},options:(null as any)}],
        options:{}},{type:7,name:'expansionHeight',definitions:[{type:0,name:'collapsed',
        styles:{type:6,styles:{height:'{{collapsedHeight}}'},offset:(null as any)},
        options:{params:{collapsedHeight:'48px'}}},{type:0,name:'expanded',styles:{type:6,
        styles:{height:'{{expandedHeight}}'},offset:(null as any)},options:{params:{expandedHeight:'64px'}}},
        {type:1,expr:'expanded <=> collapsed',animation:{type:4,styles:(null as any),
            timings:'225ms cubic-bezier(0.4,0.0,0.2,1)'},options:(null as any)}],options:{}}]}});
function View_MdExpansionPanelHeader_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),0,'span',[['class',
      'mat-expansion-indicator']],[[24,'@indicatorRotate',0]],(null as any),(null as any),
      (null as any),(null as any)))],(null as any),(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co._getExpandedState();
    _ck(_v,0,0,currVal_0);
  });
}
export function View_MdExpansionPanelHeader_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(2,[(_l()(),i0.ɵeld(0,(null as any),(null as any),3,'span',[['class',
      'mat-content']],(null as any),(null as any),(null as any),(null as any),(null as any))),
      i0.ɵncd((null as any),0),i0.ɵncd((null as any),1),i0.ɵncd((null as any),2),(_l()(),
          i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_MdExpansionPanelHeader_1)),
      i0.ɵdid(16384,(null as any),0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any))],(_ck,_v) => {
    var _co:i1.MdExpansionPanelHeader = _v.component;
    const currVal_0:any = _co._showToggle();
    _ck(_v,5,0,currVal_0);
  },(null as any));
}
export function View_MdExpansionPanelHeader_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),3,'md-expansion-panel-header',
      [['class','mat-expansion-panel-header'],['role','button']],[[1,'tabindex',0],
          [1,'aria-controls',0],[1,'aria-expanded',0],[1,'aria-disabled',0],[2,'mat-expanded',
              (null as any)],[40,'@expansionHeight',0]],[[(null as any),'click'],[(null as any),
          'keyup']],(_v,en,$event) => {
        var ad:boolean = true;
        if (('click' === en)) {
          const pd_0:any = ((<any>i0.ɵnov(_v,1)._toggle()) !== false);
          ad = (pd_0 && ad);
        }
        if (('keyup' === en)) {
          const pd_1:any = ((<any>i0.ɵnov(_v,1)._keyup($event)) !== false);
          ad = (pd_1 && ad);
        }
        return ad;
      },View_MdExpansionPanelHeader_0,RenderType_MdExpansionPanelHeader)),i0.ɵdid(180224,
      (null as any),0,i1.MdExpansionPanelHeader,[i0.Renderer2,i1.MdExpansionPanel,
          i0.ElementRef,i4.FocusMonitor,i0.ChangeDetectorRef],(null as any),(null as any)),
      i0.ɵpod({collapsedHeight:0,expandedHeight:1}),i0.ɵpod({value:0,params:1})],(null as any),
      (_ck,_v) => {
        const currVal_0:any = (i0.ɵnov(_v,1).panel.disabled? (0 - 1): 0);
        const currVal_1:any = i0.ɵnov(_v,1)._getPanelId();
        const currVal_2:any = i0.ɵnov(_v,1)._isExpanded();
        const currVal_3:any = i0.ɵnov(_v,1).panel.disabled;
        const currVal_4:any = i0.ɵnov(_v,1)._isExpanded();
        const currVal_5:any = _ck(_v,3,0,i0.ɵnov(_v,1)._getExpandedState(),_ck(_v,
            2,0,i0.ɵnov(_v,1).collapsedHeight,i0.ɵnov(_v,1).expandedHeight));
        _ck(_v,0,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4,currVal_5);
      });
}
export const MdExpansionPanelHeaderNgFactory:i0.ComponentFactory<i1.MdExpansionPanelHeader> = i0.ɵccf('md-expansion-panel-header, mat-expansion-panel-header',
    i1.MdExpansionPanelHeader,View_MdExpansionPanelHeader_Host_0,{expandedHeight:'expandedHeight',
        collapsedHeight:'collapsedHeight'},{},['md-panel-title, mat-panel-title','md-panel-description, mat-panel-description',
        '*']);
