// @angular/material/datepicker/typings/index.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from '@angular/material/datepicker';
import * as i2 from '../../dialog/typings/index.ngfactory';
import * as i3 from '@angular/common';
import * as i4 from '@angular/cdk/bidi';
import * as i5 from '@angular/platform-browser';
import * as i6 from '@angular/cdk/platform';
import * as i7 from '@angular/cdk/scrolling';
import * as i8 from '@angular/cdk/a11y';
import * as i9 from '@angular/cdk/overlay';
import * as i10 from '@angular/material/dialog';
import * as i11 from '@angular/material/icon';
import * as i12 from '@angular/http';
import * as i13 from '@angular/material/core';
import * as i14 from '@angular/material/button';
import * as i15 from '@angular/cdk/portal';
import * as i16 from '../../button/typings/index.ngfactory';
import * as i17 from '../../icon/typings/index.ngfactory';
export const MdDatepickerModuleNgFactory:i0.NgModuleFactory<i1.MdDatepickerModule> = i0.ɵcmf(i1.MdDatepickerModule,
    ([] as any[]),(_l:any) => {
      return i0.ɵmod([i0.ɵmpd(512,i0.ComponentFactoryResolver,i0.ɵCodegenComponentFactoryResolver,
          [[8,[i2.MdDialogContainerNgFactory,MdDatepickerContentNgFactory]],[3,i0.ComponentFactoryResolver],
              i0.NgModuleRef]),i0.ɵmpd(4608,i3.NgLocalization,i3.NgLocaleLocalization,
          [i0.LOCALE_ID]),i0.ɵmpd(6144,i4.DIR_DOCUMENT,(null as any),[i5.DOCUMENT]),
          i0.ɵmpd(4608,i4.Directionality,i4.Directionality,[[2,i4.DIR_DOCUMENT]]),
          i0.ɵmpd(4608,i6.Platform,i6.Platform,([] as any[])),i0.ɵmpd(5120,i7.ScrollDispatcher,
              i7.SCROLL_DISPATCHER_PROVIDER_FACTORY,[[3,i7.ScrollDispatcher],i0.NgZone,
                  i6.Platform]),i0.ɵmpd(5120,i7.ViewportRuler,i7.VIEWPORT_RULER_PROVIDER_FACTORY,
              [[3,i7.ViewportRuler],i7.ScrollDispatcher]),i0.ɵmpd(4608,i8.InteractivityChecker,
              i8.InteractivityChecker,[i6.Platform]),i0.ɵmpd(4608,i8.FocusTrapFactory,
              i8.FocusTrapFactory,[i8.InteractivityChecker,i6.Platform,i0.NgZone]),
          i0.ɵmpd(136192,i8.AriaDescriber,i8.ARIA_DESCRIBER_PROVIDER_FACTORY,[[3,i8.AriaDescriber],
              i6.Platform]),i0.ɵmpd(5120,i8.LiveAnnouncer,i8.LIVE_ANNOUNCER_PROVIDER_FACTORY,
              [[3,i8.LiveAnnouncer],[2,i8.LIVE_ANNOUNCER_ELEMENT_TOKEN],i6.Platform]),
          i0.ɵmpd(5120,i8.FocusMonitor,i8.FOCUS_MONITOR_PROVIDER_FACTORY,[[3,i8.FocusMonitor],
              i0.NgZone,i6.Platform]),i0.ɵmpd(4608,i9.ScrollStrategyOptions,i9.ScrollStrategyOptions,
              [i7.ScrollDispatcher,i7.ViewportRuler]),i0.ɵmpd(5120,i9.OverlayContainer,
              i9.ɵa,[[3,i9.OverlayContainer]]),i0.ɵmpd(4608,i9.ɵf,i9.ɵf,[i7.ViewportRuler]),
          i0.ɵmpd(4608,i9.Overlay,i9.Overlay,[i9.ScrollStrategyOptions,i9.OverlayContainer,
              i0.ComponentFactoryResolver,i9.ɵf,i0.ApplicationRef,i0.Injector,i0.NgZone]),
          i0.ɵmpd(5120,i9.ɵc,i9.ɵd,[i9.Overlay]),i0.ɵmpd(5120,i10.MD_DIALOG_SCROLL_STRATEGY,
              i10.MAT_DIALOG_SCROLL_STRATEGY_PROVIDER_FACTORY,[i9.Overlay]),i0.ɵmpd(4608,
              i10.MdDialog,i10.MdDialog,[i9.Overlay,i0.Injector,[2,i3.Location],i10.MD_DIALOG_SCROLL_STRATEGY,
                  [3,i10.MdDialog]]),i0.ɵmpd(5120,i11.MdIconRegistry,i11.ICON_REGISTRY_PROVIDER_FACTORY,
              [[3,i11.MdIconRegistry],[2,i12.Http],i5.DomSanitizer]),i0.ɵmpd(4608,
              i1.MdDatepickerIntl,i1.MdDatepickerIntl,([] as any[])),i0.ɵmpd(5120,
              i1.MD_DATEPICKER_SCROLL_STRATEGY,i1.MAT_DATEPICKER_SCROLL_STRATEGY_PROVIDER_FACTORY,
              [i9.Overlay]),i0.ɵmpd(512,i3.CommonModule,i3.CommonModule,([] as any[])),
          i0.ɵmpd(512,i13.CompatibilityModule,i13.CompatibilityModule,([] as any[])),
          i0.ɵmpd(512,i4.BidiModule,i4.BidiModule,([] as any[])),i0.ɵmpd(256,i13.MATERIAL_SANITY_CHECKS,
              true,([] as any[])),i0.ɵmpd(512,i13.MdCommonModule,i13.MdCommonModule,
              [[2,i5.DOCUMENT],[2,i13.MATERIAL_SANITY_CHECKS]]),i0.ɵmpd(512,i6.PlatformModule,
              i6.PlatformModule,([] as any[])),i0.ɵmpd(512,i7.ScrollDispatchModule,
              i7.ScrollDispatchModule,([] as any[])),i0.ɵmpd(512,i13.MdRippleModule,
              i13.MdRippleModule,([] as any[])),i0.ɵmpd(512,i8.A11yModule,i8.A11yModule,
              ([] as any[])),i0.ɵmpd(512,i14.MdButtonModule,i14.MdButtonModule,([] as any[])),
          i0.ɵmpd(512,i15.PortalModule,i15.PortalModule,([] as any[])),i0.ɵmpd(512,
              i9.OverlayModule,i9.OverlayModule,([] as any[])),i0.ɵmpd(512,i10.MdDialogModule,
              i10.MdDialogModule,([] as any[])),i0.ɵmpd(512,i11.MdIconModule,i11.MdIconModule,
              ([] as any[])),i0.ɵmpd(512,i1.MdDatepickerModule,i1.MdDatepickerModule,
              ([] as any[]))]);
    });
const styles_MdCalendar:any[] = ['.mat-calendar{display:block}.mat-calendar-header{padding:8px 8px 0 8px}.mat-calendar-content{padding:0 8px 8px 8px;outline:0}.mat-calendar-controls{display:flex;margin:5% calc(33% / 7 - 16px)}.mat-calendar-spacer{flex:1 1 auto}.mat-calendar-period-button{min-width:0}.mat-calendar-arrow{display:inline-block;width:0;height:0;border-left:5px solid transparent;border-right:5px solid transparent;border-top-width:5px;border-top-style:solid;margin:0 0 0 5px;vertical-align:middle}.mat-calendar-arrow.mat-calendar-invert{transform:rotate(180deg)}[dir=rtl] .mat-calendar-arrow{margin:0 5px 0 0}.mat-calendar-next-button,.mat-calendar-previous-button{position:relative}.mat-calendar-next-button::after,.mat-calendar-previous-button::after{top:0;left:0;right:0;bottom:0;position:absolute;content:\'\';margin:15.5px;border:0 solid currentColor;border-top-width:2px}[dir=rtl] .mat-calendar-next-button,[dir=rtl] .mat-calendar-previous-button{transform:rotate(180deg)}.mat-calendar-previous-button::after{border-left-width:2px;transform:translateX(2px) rotate(-45deg)}.mat-calendar-next-button::after{border-right-width:2px;transform:translateX(-2px) rotate(45deg)}.mat-calendar-table{border-spacing:0;border-collapse:collapse;width:100%}.mat-calendar-table-header th{text-align:center;padding:0 0 8px 0}.mat-calendar-table-header-divider{position:relative;height:1px}.mat-calendar-table-header-divider::after{content:\'\';position:absolute;top:0;left:-8px;right:-8px;height:1px}'];
export const RenderType_MdCalendar:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_MdCalendar,
    data:{}});
function View_MdCalendar_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),2,'md-month-view',
      ([] as any[]),(null as any),[[(null as any),'selectedChange'],[(null as any),
          'userSelection']],(_v,en,$event) => {
        var ad:boolean = true;
        var _co:any = _v.component;
        if (('selectedChange' === en)) {
          const pd_0:any = ((<any>_co._dateSelected($event)) !== false);
          ad = (pd_0 && ad);
        }
        if (('userSelection' === en)) {
          const pd_1:any = ((<any>_co._userSelected()) !== false);
          ad = (pd_1 && ad);
        }
        return ad;
      },View_MdMonthView_0,RenderType_MdMonthView)),i0.ɵdid(1097728,(null as any),
      0,i1.MdMonthView,[[2,i13.DateAdapter],[2,i13.MD_DATE_FORMATS]],{activeDate:[0,
          'activeDate'],selected:[1,'selected'],dateFilter:[2,'dateFilter']},{selectedChange:'selectedChange',
          userSelection:'userSelection'}),i0.ɵprd(8448,(null as any),i13.MATERIAL_COMPATIBILITY_MODE,
      true,([] as any[]))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co._activeDate;
    const currVal_1:any = _co.selected;
    const currVal_2:any = _co._dateFilterForViews;
    _ck(_v,1,0,currVal_0,currVal_1,currVal_2);
  },(null as any));
}
function View_MdCalendar_2(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'mat-year-view',
      ([] as any[]),(null as any),[[(null as any),'selectedChange']],(_v,en,$event) => {
        var ad:boolean = true;
        var _co:any = _v.component;
        if (('selectedChange' === en)) {
          const pd_0:any = ((<any>_co._monthSelected($event)) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },View_MdYearView_0,RenderType_MdYearView)),i0.ɵdid(1097728,(null as any),0,
      i1.MdYearView,[[2,i13.DateAdapter],[2,i13.MD_DATE_FORMATS]],{activeDate:[0,'activeDate'],
          selected:[1,'selected'],dateFilter:[2,'dateFilter']},{selectedChange:'selectedChange'})],
      (_ck,_v) => {
        var _co:any = _v.component;
        const currVal_0:any = _co._activeDate;
        const currVal_1:any = _co.selected;
        const currVal_2:any = _co._dateFilterForViews;
        _ck(_v,1,0,currVal_0,currVal_1,currVal_2);
      },(null as any));
}
export function View_MdCalendar_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(2,[(_l()(),i0.ɵeld(0,(null as any),(null as any),19,'div',[['class',
      'mat-calendar-header']],(null as any),(null as any),(null as any),(null as any),
      (null as any))),(_l()(),i0.ɵeld(0,(null as any),(null as any),18,'div',[['class',
      'mat-calendar-controls']],(null as any),(null as any),(null as any),(null as any),
      (null as any))),(_l()(),i0.ɵeld(0,(null as any),(null as any),6,'button',[['class',
      'mat-calendar-period-button mat-button'],['mat-button','']],[[1,'aria-label',
      0],[8,'disabled',0]],[[(null as any),'click']],(_v,en,$event) => {
    var ad:boolean = true;
    var _co:i1.MdCalendar<any> = _v.component;
    if (('click' === en)) {
      const pd_0:any = ((<any>_co._currentPeriodClicked()) !== false);
      ad = (pd_0 && ad);
    }
    return ad;
  },i16.View_MdButton_0,i16.RenderType_MdButton)),i0.ɵdid(16384,(null as any),0,i13.MatPrefixRejector,
      [[2,i13.MATERIAL_COMPATIBILITY_MODE],i0.ElementRef],(null as any),(null as any)),
      i0.ɵdid(180224,(null as any),0,i14.MdButton,[i0.Renderer2,i0.ElementRef,i6.Platform,
          i8.FocusMonitor],(null as any),(null as any)),i0.ɵdid(16384,(null as any),
          0,i14.MdButtonCssMatStyler,([] as any[]),(null as any),(null as any)),i0.ɵprd(8448,
          (null as any),i13.MATERIAL_COMPATIBILITY_MODE,true,([] as any[])),(_l()(),
          i0.ɵted(0,['',''])),(_l()(),i0.ɵeld(0,(null as any),0,0,'div',[['class',
          'mat-calendar-arrow']],[[2,'mat-calendar-invert',(null as any)]],(null as any),
          (null as any),(null as any),(null as any))),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),0,'div',[['class','mat-calendar-spacer']],(null as any),(null as any),
          (null as any),(null as any),(null as any))),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),4,'button',[['class','mat-calendar-previous-button mat-icon-button'],
              ['mat-icon-button','']],[[1,'aria-label',0],[8,'disabled',0]],[[(null as any),
              'click']],(_v,en,$event) => {
            var ad:boolean = true;
            var _co:i1.MdCalendar<any> = _v.component;
            if (('click' === en)) {
              const pd_0:any = ((<any>_co._previousClicked()) !== false);
              ad = (pd_0 && ad);
            }
            return ad;
          },i16.View_MdButton_0,i16.RenderType_MdButton)),i0.ɵdid(16384,(null as any),
          0,i13.MatPrefixRejector,[[2,i13.MATERIAL_COMPATIBILITY_MODE],i0.ElementRef],
          (null as any),(null as any)),i0.ɵdid(180224,(null as any),0,i14.MdButton,
          [i0.Renderer2,i0.ElementRef,i6.Platform,i8.FocusMonitor],{disabled:[0,'disabled']},
          (null as any)),i0.ɵdid(16384,(null as any),0,i14.MdIconButtonCssMatStyler,
          ([] as any[]),(null as any),(null as any)),i0.ɵprd(8448,(null as any),i13.MATERIAL_COMPATIBILITY_MODE,
          true,([] as any[])),(_l()(),i0.ɵeld(0,(null as any),(null as any),4,'button',
          [['class','mat-calendar-next-button mat-icon-button'],['mat-icon-button',
              '']],[[1,'aria-label',0],[8,'disabled',0]],[[(null as any),'click']],
          (_v,en,$event) => {
            var ad:boolean = true;
            var _co:i1.MdCalendar<any> = _v.component;
            if (('click' === en)) {
              const pd_0:any = ((<any>_co._nextClicked()) !== false);
              ad = (pd_0 && ad);
            }
            return ad;
          },i16.View_MdButton_0,i16.RenderType_MdButton)),i0.ɵdid(16384,(null as any),
          0,i13.MatPrefixRejector,[[2,i13.MATERIAL_COMPATIBILITY_MODE],i0.ElementRef],
          (null as any),(null as any)),i0.ɵdid(180224,(null as any),0,i14.MdButton,
          [i0.Renderer2,i0.ElementRef,i6.Platform,i8.FocusMonitor],{disabled:[0,'disabled']},
          (null as any)),i0.ɵdid(16384,(null as any),0,i14.MdIconButtonCssMatStyler,
          ([] as any[]),(null as any),(null as any)),i0.ɵprd(8448,(null as any),i13.MATERIAL_COMPATIBILITY_MODE,
          true,([] as any[])),(_l()(),i0.ɵeld(0,(null as any),(null as any),6,'div',
          [['cdkMonitorSubtreeFocus',''],['class','mat-calendar-content']],(null as any),
          [[(null as any),'keydown']],(_v,en,$event) => {
            var ad:boolean = true;
            var _co:i1.MdCalendar<any> = _v.component;
            if (('keydown' === en)) {
              const pd_0:any = ((<any>_co._handleCalendarBodyKeydown($event)) !== false);
              ad = (pd_0 && ad);
            }
            return ad;
          },(null as any),(null as any))),i0.ɵdid(16384,(null as any),0,i3.NgSwitch,
          ([] as any[]),{ngSwitch:[0,'ngSwitch']},(null as any)),i0.ɵdid(147456,(null as any),
          0,i8.CdkMonitorFocus,[i0.ElementRef,i8.FocusMonitor,i0.Renderer2],(null as any),
          (null as any)),(_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),
          View_MdCalendar_1)),i0.ɵdid(278528,(null as any),0,i3.NgSwitchCase,[i0.ViewContainerRef,
          i0.TemplateRef,i3.NgSwitch],{ngSwitchCase:[0,'ngSwitchCase']},(null as any)),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_MdCalendar_2)),
      i0.ɵdid(16384,(null as any),0,i3.NgSwitchDefault,[i0.ViewContainerRef,i0.TemplateRef,
          i3.NgSwitch],(null as any),(null as any))],(_ck,_v) => {
    var _co:i1.MdCalendar<any> = _v.component;
    const currVal_6:boolean = !_co._previousEnabled();
    _ck(_v,12,0,currVal_6);
    const currVal_9:boolean = !_co._nextEnabled();
    _ck(_v,17,0,currVal_9);
    const currVal_10:any = _co._monthView;
    _ck(_v,21,0,currVal_10);
    const currVal_11:any = true;
    _ck(_v,24,0,currVal_11);
  },(_ck,_v) => {
    var _co:i1.MdCalendar<any> = _v.component;
    const currVal_0:any = _co._periodButtonLabel;
    const currVal_1:any = (i0.ɵnov(_v,4).disabled || (null as any));
    _ck(_v,2,0,currVal_0,currVal_1);
    const currVal_2:any = _co._periodButtonText;
    _ck(_v,7,0,currVal_2);
    const currVal_3:boolean = !_co._monthView;
    _ck(_v,8,0,currVal_3);
    const currVal_4:any = _co._prevButtonLabel;
    const currVal_5:any = (i0.ɵnov(_v,12).disabled || (null as any));
    _ck(_v,10,0,currVal_4,currVal_5);
    const currVal_7:any = _co._nextButtonLabel;
    const currVal_8:any = (i0.ɵnov(_v,17).disabled || (null as any));
    _ck(_v,15,0,currVal_7,currVal_8);
  });
}
export function View_MdCalendar_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),2,'md-calendar',
      [['class','mat-calendar']],(null as any),(null as any),(null as any),View_MdCalendar_0,
      RenderType_MdCalendar)),i0.ɵdid(1228800,(null as any),0,i1.MdCalendar,[i0.ElementRef,
      i1.MdDatepickerIntl,i0.NgZone,[2,i13.DateAdapter],[2,i13.MD_DATE_FORMATS],i0.ChangeDetectorRef],
      (null as any),(null as any)),i0.ɵprd(8448,(null as any),i13.MATERIAL_COMPATIBILITY_MODE,
      true,([] as any[]))],(null as any),(null as any));
}
export const MdCalendarNgFactory:i0.ComponentFactory<i1.MdCalendar<any>> = i0.ɵccf('md-calendar, mat-calendar',
    i1.MdCalendar,View_MdCalendar_Host_0,{startAt:'startAt',startView:'startView',
        selected:'selected',minDate:'minDate',maxDate:'maxDate',dateFilter:'dateFilter'},
    {selectedChange:'selectedChange',userSelection:'userSelection'},([] as any[]));
const styles_MdCalendarBody:any[] = ['.mat-calendar-body{min-width:224px}.mat-calendar-body-label{height:0;line-height:0;text-align:left;padding-left:4.71429%;padding-right:4.71429%}.mat-calendar-body-cell{position:relative;height:0;line-height:0;text-align:center;outline:0;cursor:pointer}.mat-calendar-body-disabled{cursor:default}.mat-calendar-body-cell-content{position:absolute;top:5%;left:5%;display:flex;align-items:center;justify-content:center;box-sizing:border-box;width:90%;height:90%;line-height:1;border-width:1px;border-style:solid;border-radius:999px}[dir=rtl] .mat-calendar-body-label{text-align:right}'];
export const RenderType_MdCalendarBody:i0.RendererType2 = i0.ɵcrt({encapsulation:2,
    styles:styles_MdCalendarBody,data:{}});
function View_MdCalendarBody_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),2,'tr',[['aria-hidden',
      'true']],(null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),1,'td',[['class','mat-calendar-body-label']],
          [[1,'colspan',0],[4,'paddingTop','%'],[4,'paddingBottom','%']],(null as any),
          (null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),
          ['','']))],(null as any),(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.numCols;
    const currVal_1:any = ((50 * _co.cellAspectRatio) / _co.numCols);
    const currVal_2:any = ((50 * _co.cellAspectRatio) / _co.numCols);
    _ck(_v,1,0,currVal_0,currVal_1,currVal_2);
    const currVal_3:any = _co.label;
    _ck(_v,2,0,currVal_3);
  });
}
function View_MdCalendarBody_3(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'td',[['aria-hidden',
      'true'],['class','mat-calendar-body-label']],[[1,'colspan',0],[4,'paddingTop',
      '%'],[4,'paddingBottom','%']],(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['','']))],(null as any),(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co._firstRowOffset;
    const currVal_1:any = ((50 * _co.cellAspectRatio) / _co.numCols);
    const currVal_2:any = ((50 * _co.cellAspectRatio) / _co.numCols);
    _ck(_v,0,0,currVal_0,currVal_1,currVal_2);
    const currVal_3:any = ((_co._firstRowOffset >= _co.labelMinRequiredCells)? _co.label: '');
    _ck(_v,1,0,currVal_3);
  });
}
function View_MdCalendarBody_4(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),2,'td',[['class',
      'mat-calendar-body-cell'],['role','gridcell']],[[8,'tabIndex',0],[2,'mat-calendar-body-disabled',
      (null as any)],[2,'mat-calendar-body-active',(null as any)],[1,'aria-label',
      0],[1,'aria-disabled',0],[4,'width','%'],[4,'paddingTop','%'],[4,'paddingBottom',
      '%']],[[(null as any),'click']],(_v,en,$event) => {
    var ad:boolean = true;
    var _co:any = _v.component;
    if (('click' === en)) {
      const pd_0:any = ((<any>_co._cellClicked(_v.context.$implicit)) !== false);
      ad = (pd_0 && ad);
    }
    return ad;
  },(null as any),(null as any))),(_l()(),i0.ɵeld(0,(null as any),(null as any),1,
      'div',[['class','mat-calendar-body-cell-content']],[[2,'mat-calendar-body-selected',
          (null as any)],[2,'mat-calendar-body-today',(null as any)]],(null as any),
      (null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['',
      '']))],(null as any),(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = (_co._isActiveCell((<any>_v.parent).context.index,_v.context.index)? 0: (0 - 1));
    const currVal_1:boolean = !_v.context.$implicit.enabled;
    const currVal_2:any = _co._isActiveCell((<any>_v.parent).context.index,_v.context.index);
    const currVal_3:any = _v.context.$implicit.ariaLabel;
    const currVal_4:boolean = (!_v.context.$implicit.enabled || (null as any));
    const currVal_5:any = (100 / _co.numCols);
    const currVal_6:any = ((50 * _co.cellAspectRatio) / _co.numCols);
    const currVal_7:any = ((50 * _co.cellAspectRatio) / _co.numCols);
    _ck(_v,0,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4,currVal_5,currVal_6,
        currVal_7);
    const currVal_8:any = (_co.selectedValue === _v.context.$implicit.value);
    const currVal_9:any = (_co.todayValue === _v.context.$implicit.value);
    _ck(_v,1,0,currVal_8,currVal_9);
    const currVal_10:any = _v.context.$implicit.displayValue;
    _ck(_v,2,0,currVal_10);
  });
}
function View_MdCalendarBody_2(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),4,'tr',[['role',
      'row']],(null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_MdCalendarBody_3)),
      i0.ɵdid(16384,(null as any),0,i3.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵand(16777216,(null as any),(null as any),
          1,(null as any),View_MdCalendarBody_4)),i0.ɵdid(802816,(null as any),0,i3.NgForOf,
          [i0.ViewContainerRef,i0.TemplateRef,i0.IterableDiffers],{ngForOf:[0,'ngForOf']},
          (null as any))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = ((_v.context.index === 0) && _co._firstRowOffset);
    _ck(_v,2,0,currVal_0);
    const currVal_1:any = _v.context.$implicit;
    _ck(_v,4,0,currVal_1);
  },(null as any));
}
export function View_MdCalendarBody_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(2,[(_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),
      View_MdCalendarBody_1)),i0.ɵdid(16384,(null as any),0,i3.NgIf,[i0.ViewContainerRef,
      i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),(_l()(),i0.ɵand(16777216,(null as any),
      (null as any),1,(null as any),View_MdCalendarBody_2)),i0.ɵdid(802816,(null as any),
      0,i3.NgForOf,[i0.ViewContainerRef,i0.TemplateRef,i0.IterableDiffers],{ngForOf:[0,
          'ngForOf']},(null as any))],(_ck,_v) => {
    var _co:i1.MdCalendarBody = _v.component;
    const currVal_0:any = (_co._firstRowOffset < _co.labelMinRequiredCells);
    _ck(_v,1,0,currVal_0);
    const currVal_1:any = _co.rows;
    _ck(_v,3,0,currVal_1);
  },(null as any));
}
export function View_MdCalendarBody_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'div',[['class',
      'mat-calendar-body'],['md-calendar-body','']],(null as any),(null as any),(null as any),
      View_MdCalendarBody_0,RenderType_MdCalendarBody)),i0.ɵdid(49152,(null as any),
      0,i1.MdCalendarBody,([] as any[]),(null as any),(null as any))],(null as any),
      (null as any));
}
export const MdCalendarBodyNgFactory:i0.ComponentFactory<i1.MdCalendarBody> = i0.ɵccf('[md-calendar-body], [mat-calendar-body]',
    i1.MdCalendarBody,View_MdCalendarBody_Host_0,{label:'label',rows:'rows',todayValue:'todayValue',
        selectedValue:'selectedValue',labelMinRequiredCells:'labelMinRequiredCells',
        numCols:'numCols',allowDisabledSelection:'allowDisabledSelection',activeCell:'activeCell',
        cellAspectRatio:'cellAspectRatio'},{selectedValueChange:'selectedValueChange'},
    ([] as any[]));
const styles_MdDatepicker:any[] = ([] as any[]);
export const RenderType_MdDatepicker:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_MdDatepicker,
    data:{}});
export function View_MdDatepicker_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(2,([] as any[]),(null as any),(null as any));
}
export function View_MdDatepicker_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(16777216,(null as any),(null as any),1,'md-datepicker',
      ([] as any[]),(null as any),(null as any),(null as any),View_MdDatepicker_0,
      RenderType_MdDatepicker)),i0.ɵdid(180224,(null as any),0,i1.MdDatepicker,[i10.MdDialog,
      i9.Overlay,i0.NgZone,i0.ViewContainerRef,i1.MD_DATEPICKER_SCROLL_STRATEGY,[2,
          i13.DateAdapter],[2,i4.Directionality],[2,i5.DOCUMENT]],(null as any),(null as any))],
      (null as any),(null as any));
}
export const MdDatepickerNgFactory:i0.ComponentFactory<i1.MdDatepicker<any>> = i0.ɵccf('md-datepicker, mat-datepicker',
    i1.MdDatepicker,View_MdDatepicker_Host_0,{startAt:'startAt',startView:'startView',
        touchUi:'touchUi',disabled:'disabled'},{selectedChanged:'selectedChanged'},
    ([] as any[]));
const styles_MdDatepickerContent:any[] = ['.mat-datepicker-content{box-shadow:0 5px 5px -3px rgba(0,0,0,.2),0 8px 10px 1px rgba(0,0,0,.14),0 3px 14px 2px rgba(0,0,0,.12);display:block}.mat-calendar{width:296px;height:354px}.mat-datepicker-content-touch{box-shadow:0 0 0 0 rgba(0,0,0,.2),0 0 0 0 rgba(0,0,0,.14),0 0 0 0 rgba(0,0,0,.12);display:block;max-height:80vh;overflow:auto;margin:-24px}.mat-datepicker-content-touch .mat-calendar{min-width:250px;min-height:312px;max-width:750px;max-height:788px}@media all and (orientation:landscape){.mat-datepicker-content-touch .mat-calendar{width:64vh;height:80vh}}@media all and (orientation:portrait){.mat-datepicker-content-touch .mat-calendar{width:80vw;height:100vw}}'];
export const RenderType_MdDatepickerContent:i0.RendererType2 = i0.ɵcrt({encapsulation:2,
    styles:styles_MdDatepickerContent,data:{}});
export function View_MdDatepickerContent_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(2,[i0.ɵqud(402653184,1,{_calendar:0}),(_l()(),i0.ɵeld(0,(null as any),
      (null as any),3,'mat-calendar',[['cdkTrapFocus',''],['class','mat-calendar']],
      [[8,'id',0]],[[(null as any),'selectedChange'],[(null as any),'userSelection']],
      (_v,en,$event) => {
        var ad:boolean = true;
        var _co:i1.MdDatepickerContent<any> = _v.component;
        if (('selectedChange' === en)) {
          const pd_0:any = ((<any>_co.datepicker._select($event)) !== false);
          ad = (pd_0 && ad);
        }
        if (('userSelection' === en)) {
          const pd_1:any = ((<any>_co.datepicker.close()) !== false);
          ad = (pd_1 && ad);
        }
        return ad;
      },View_MdCalendar_0,RenderType_MdCalendar)),i0.ɵdid(1196032,(null as any),0,
      i8.FocusTrapDirective,[i0.ElementRef,i8.FocusTrapFactory],{enabled:[0,'enabled']},
      (null as any)),i0.ɵdid(1228800,[[1,4]],0,i1.MdCalendar,[i0.ElementRef,i1.MdDatepickerIntl,
      i0.NgZone,[2,i13.DateAdapter],[2,i13.MD_DATE_FORMATS],i0.ChangeDetectorRef],
      {startAt:[0,'startAt'],startView:[1,'startView'],selected:[2,'selected'],minDate:[3,
          'minDate'],maxDate:[4,'maxDate'],dateFilter:[5,'dateFilter']},{selectedChange:'selectedChange',
          userSelection:'userSelection'}),i0.ɵprd(8448,(null as any),i13.MATERIAL_COMPATIBILITY_MODE,
      true,([] as any[]))],(_ck,_v) => {
    var _co:i1.MdDatepickerContent<any> = _v.component;
    const currVal_1:any = '';
    _ck(_v,2,0,currVal_1);
    const currVal_2:any = _co.datepicker.startAt;
    const currVal_3:any = _co.datepicker.startView;
    const currVal_4:any = _co.datepicker._selected;
    const currVal_5:any = _co.datepicker._minDate;
    const currVal_6:any = _co.datepicker._maxDate;
    const currVal_7:any = _co.datepicker._dateFilter;
    _ck(_v,3,0,currVal_2,currVal_3,currVal_4,currVal_5,currVal_6,currVal_7);
  },(_ck,_v) => {
    var _co:i1.MdDatepickerContent<any> = _v.component;
    const currVal_0:any = _co.datepicker.id;
    _ck(_v,1,0,currVal_0);
  });
}
export function View_MdDatepickerContent_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),2,'md-datepicker-content',
      [['class','mat-datepicker-content']],[[2,'mat-datepicker-content-touch',(null as any)]],
      [[(null as any),'keydown']],(_v,en,$event) => {
        var ad:boolean = true;
        if (('keydown' === en)) {
          const pd_0:any = ((<any>i0.ɵnov(_v,1)._handleKeydown($event)) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },View_MdDatepickerContent_0,RenderType_MdDatepickerContent)),i0.ɵdid(1097728,
      (null as any),0,i1.MdDatepickerContent,([] as any[]),(null as any),(null as any)),
      i0.ɵprd(8448,(null as any),i13.MATERIAL_COMPATIBILITY_MODE,true,([] as any[]))],
      (null as any),(_ck,_v) => {
        const currVal_0:any = i0.ɵnov(_v,1).datepicker.touchUi;
        _ck(_v,0,0,currVal_0);
      });
}
export const MdDatepickerContentNgFactory:i0.ComponentFactory<i1.MdDatepickerContent<any>> = i0.ɵccf('md-datepicker-content, mat-datepicker-content',
    i1.MdDatepickerContent,View_MdDatepickerContent_Host_0,{},{},([] as any[]));
const styles_MdDatepickerToggle:any[] = ([] as any[]);
export const RenderType_MdDatepickerToggle:i0.RendererType2 = i0.ɵcrt({encapsulation:2,
    styles:styles_MdDatepickerToggle,data:{}});
export function View_MdDatepickerToggle_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(2,[(_l()(),i0.ɵeld(0,(null as any),(null as any),10,'button',[['class',
      'mat-icon-button'],['mat-icon-button',''],['type','button']],[[1,'aria-label',
      0],[8,'disabled',0]],[[(null as any),'click']],(_v,en,$event) => {
    var ad:boolean = true;
    var _co:i1.MdDatepickerToggle<any> = _v.component;
    if (('click' === en)) {
      const pd_0:any = ((<any>_co._open($event)) !== false);
      ad = (pd_0 && ad);
    }
    return ad;
  },i16.View_MdButton_0,i16.RenderType_MdButton)),i0.ɵdid(16384,(null as any),0,i13.MatPrefixRejector,
      [[2,i13.MATERIAL_COMPATIBILITY_MODE],i0.ElementRef],(null as any),(null as any)),
      i0.ɵdid(180224,(null as any),0,i14.MdButton,[i0.Renderer2,i0.ElementRef,i6.Platform,
          i8.FocusMonitor],{disabled:[0,'disabled']},(null as any)),i0.ɵdid(16384,
          (null as any),0,i14.MdIconButtonCssMatStyler,([] as any[]),(null as any),
          (null as any)),i0.ɵprd(8448,(null as any),i13.MATERIAL_COMPATIBILITY_MODE,
          true,([] as any[])),(_l()(),i0.ɵeld(0,(null as any),0,5,'mat-icon',[['class',
          'mat-icon'],['role','img']],(null as any),(null as any),(null as any),i17.View_MdIcon_0,
          i17.RenderType_MdIcon)),i0.ɵdid(16384,(null as any),0,i13.MatPrefixRejector,
          [[2,i13.MATERIAL_COMPATIBILITY_MODE],i0.ElementRef],(null as any),(null as any)),
      i0.ɵdid(638976,(null as any),0,i11.MdIcon,[i0.Renderer2,i0.ElementRef,i11.MdIconRegistry,
          [8,(null as any)]],(null as any),(null as any)),(_l()(),i0.ɵeld(0,(null as any),
          0,2,':svg:svg',[['fill','currentColor'],['focusable','false'],['height',
              '100%'],['style','vertical-align: top'],['viewBox','0 0 24 24'],['width',
              '100%'],['xmlns','http://www.w3.org/2000/svg']],(null as any),(null as any),
          (null as any),(null as any),(null as any))),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),0,':svg:path',[['d','M0 0h24v24H0z'],['fill','none']],(null as any),
          (null as any),(null as any),(null as any),(null as any))),(_l()(),i0.ɵeld(0,
          (null as any),(null as any),0,':svg:path',[['d','M19 3h-1V1h-2v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 16H5V8h14v11zM7 10h5v5H7z']],
          (null as any),(null as any),(null as any),(null as any),(null as any)))],
      (_ck,_v) => {
        var _co:i1.MdDatepickerToggle<any> = _v.component;
        const currVal_2:any = _co.disabled;
        _ck(_v,2,0,currVal_2);
        _ck(_v,7,0);
      },(_ck,_v) => {
        var _co:i1.MdDatepickerToggle<any> = _v.component;
        const currVal_0:any = _co._intl.openCalendarLabel;
        const currVal_1:any = (i0.ɵnov(_v,2).disabled || (null as any));
        _ck(_v,0,0,currVal_0,currVal_1);
      });
}
export function View_MdDatepickerToggle_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),2,'md-datepicker-toggle',
      [['class','mat-datepicker-toggle']],(null as any),(null as any),(null as any),
      View_MdDatepickerToggle_0,RenderType_MdDatepickerToggle)),i0.ɵdid(704512,(null as any),
      0,i1.MdDatepickerToggle,[i1.MdDatepickerIntl,i0.ChangeDetectorRef],(null as any),
      (null as any)),i0.ɵprd(8448,(null as any),i13.MATERIAL_COMPATIBILITY_MODE,true,
      ([] as any[]))],(null as any),(null as any));
}
export const MdDatepickerToggleNgFactory:i0.ComponentFactory<i1.MdDatepickerToggle<any>> = i0.ɵccf('md-datepicker-toggle, mat-datepicker-toggle',
    i1.MdDatepickerToggle,View_MdDatepickerToggle_Host_0,{datepicker:'for',disabled:'disabled'},
    {},([] as any[]));
const styles_MdMonthView:any[] = ([] as any[]);
export const RenderType_MdMonthView:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_MdMonthView,
    data:{}});
function View_MdMonthView_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'th',([] as any[]),
      [[1,'aria-label',0]],(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['','']))],(null as any),(_ck,_v) => {
    const currVal_0:any = _v.context.$implicit.long;
    _ck(_v,0,0,currVal_0);
    const currVal_1:any = _v.context.$implicit.narrow;
    _ck(_v,1,0,currVal_1);
  });
}
export function View_MdMonthView_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(2,[(_l()(),i0.ɵeld(0,(null as any),(null as any),8,'table',[['class',
      'mat-calendar-table']],(null as any),(null as any),(null as any),(null as any),
      (null as any))),(_l()(),i0.ɵeld(0,(null as any),(null as any),5,'thead',[['class',
      'mat-calendar-table-header']],(null as any),(null as any),(null as any),(null as any),
      (null as any))),(_l()(),i0.ɵeld(0,(null as any),(null as any),2,'tr',([] as any[]),
      (null as any),(null as any),(null as any),(null as any),(null as any))),(_l()(),
      i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_MdMonthView_1)),
      i0.ɵdid(802816,(null as any),0,i3.NgForOf,[i0.ViewContainerRef,i0.TemplateRef,
          i0.IterableDiffers],{ngForOf:[0,'ngForOf']},(null as any)),(_l()(),i0.ɵeld(0,
          (null as any),(null as any),1,'tr',([] as any[]),(null as any),(null as any),
          (null as any),(null as any),(null as any))),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),0,'th',[['aria-hidden','true'],['class','mat-calendar-table-header-divider'],
              ['colspan','7']],(null as any),(null as any),(null as any),(null as any),
          (null as any))),(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'tbody',
          [['class','mat-calendar-body'],['mat-calendar-body',''],['role','grid']],
          (null as any),[[(null as any),'selectedValueChange']],(_v,en,$event) => {
            var ad:boolean = true;
            var _co:i1.MdMonthView<any> = _v.component;
            if (('selectedValueChange' === en)) {
              const pd_0:any = ((<any>_co._dateSelected($event)) !== false);
              ad = (pd_0 && ad);
            }
            return ad;
          },View_MdCalendarBody_0,RenderType_MdCalendarBody)),i0.ɵdid(49152,(null as any),
          0,i1.MdCalendarBody,([] as any[]),{label:[0,'label'],rows:[1,'rows'],todayValue:[2,
              'todayValue'],selectedValue:[3,'selectedValue'],labelMinRequiredCells:[4,
              'labelMinRequiredCells'],activeCell:[5,'activeCell']},{selectedValueChange:'selectedValueChange'})],
      (_ck,_v) => {
        var _co:i1.MdMonthView<any> = _v.component;
        const currVal_0:any = _co._weekdays;
        _ck(_v,4,0,currVal_0);
        const currVal_1:any = _co._monthLabel;
        const currVal_2:any = _co._weeks;
        const currVal_3:any = _co._todayDate;
        const currVal_4:any = _co._selectedDate;
        const currVal_5:any = 3;
        const currVal_6:any = (_co._dateAdapter.getDate(_co.activeDate) - 1);
        _ck(_v,8,0,currVal_1,currVal_2,currVal_3,currVal_4,currVal_5,currVal_6);
      },(null as any));
}
export function View_MdMonthView_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),2,'md-month-view',
      ([] as any[]),(null as any),(null as any),(null as any),View_MdMonthView_0,RenderType_MdMonthView)),
      i0.ɵdid(1097728,(null as any),0,i1.MdMonthView,[[2,i13.DateAdapter],[2,i13.MD_DATE_FORMATS]],
          (null as any),(null as any)),i0.ɵprd(8448,(null as any),i13.MATERIAL_COMPATIBILITY_MODE,
          true,([] as any[]))],(null as any),(null as any));
}
export const MdMonthViewNgFactory:i0.ComponentFactory<i1.MdMonthView<any>> = i0.ɵccf('md-month-view',
    i1.MdMonthView,View_MdMonthView_Host_0,{activeDate:'activeDate',selected:'selected',
        dateFilter:'dateFilter'},{selectedChange:'selectedChange',userSelection:'userSelection'},
    ([] as any[]));
const styles_MdYearView:any[] = ([] as any[]);
export const RenderType_MdYearView:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_MdYearView,
    data:{}});
export function View_MdYearView_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(2,[(_l()(),i0.ɵeld(0,(null as any),(null as any),5,'table',[['class',
      'mat-calendar-table']],(null as any),(null as any),(null as any),(null as any),
      (null as any))),(_l()(),i0.ɵeld(0,(null as any),(null as any),2,'thead',[['class',
      'mat-calendar-table-header']],(null as any),(null as any),(null as any),(null as any),
      (null as any))),(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'tr',([] as any[]),
      (null as any),(null as any),(null as any),(null as any),(null as any))),(_l()(),
      i0.ɵeld(0,(null as any),(null as any),0,'th',[['class','mat-calendar-table-header-divider'],
          ['colspan','4']],(null as any),(null as any),(null as any),(null as any),
          (null as any))),(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'tbody',
      [['allowDisabledSelection','true'],['class','mat-calendar-body'],['mat-calendar-body',
          ''],['role','grid']],(null as any),[[(null as any),'selectedValueChange']],
      (_v,en,$event) => {
        var ad:boolean = true;
        var _co:i1.MdYearView<any> = _v.component;
        if (('selectedValueChange' === en)) {
          const pd_0:any = ((<any>_co._monthSelected($event)) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },View_MdCalendarBody_0,RenderType_MdCalendarBody)),i0.ɵdid(49152,(null as any),
      0,i1.MdCalendarBody,([] as any[]),{label:[0,'label'],rows:[1,'rows'],todayValue:[2,
          'todayValue'],selectedValue:[3,'selectedValue'],labelMinRequiredCells:[4,
          'labelMinRequiredCells'],numCols:[5,'numCols'],allowDisabledSelection:[6,
          'allowDisabledSelection'],activeCell:[7,'activeCell'],cellAspectRatio:[8,
          'cellAspectRatio']},{selectedValueChange:'selectedValueChange'})],(_ck,_v) => {
    var _co:i1.MdYearView<any> = _v.component;
    const currVal_0:any = _co._yearLabel;
    const currVal_1:any = _co._months;
    const currVal_2:any = _co._todayMonth;
    const currVal_3:any = _co._selectedMonth;
    const currVal_4:any = 2;
    const currVal_5:any = 4;
    const currVal_6:any = 'true';
    const currVal_7:any = _co._dateAdapter.getMonth(_co.activeDate);
    const currVal_8:any = (4 / 7);
    _ck(_v,5,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4,currVal_5,currVal_6,
        currVal_7,currVal_8);
  },(null as any));
}
export function View_MdYearView_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'md-year-view',
      ([] as any[]),(null as any),(null as any),(null as any),View_MdYearView_0,RenderType_MdYearView)),
      i0.ɵdid(1097728,(null as any),0,i1.MdYearView,[[2,i13.DateAdapter],[2,i13.MD_DATE_FORMATS]],
          (null as any),(null as any))],(null as any),(null as any));
}
export const MdYearViewNgFactory:i0.ComponentFactory<i1.MdYearView<any>> = i0.ɵccf('md-year-view, mat-year-view',
    i1.MdYearView,View_MdYearView_Host_0,{activeDate:'activeDate',selected:'selected',
        dateFilter:'dateFilter'},{selectedChange:'selectedChange'},([] as any[]));
