// @angular/material/tabs/typings/index.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from '@angular/material/tabs';
import * as i2 from '@angular/common';
import * as i3 from '@angular/cdk/bidi';
import * as i4 from '@angular/platform-browser';
import * as i5 from '@angular/cdk/platform';
import * as i6 from '@angular/cdk/scrolling';
import * as i7 from '@angular/cdk/observers';
import * as i8 from '@angular/material/core';
import * as i9 from '@angular/cdk/portal';
export const MdTabsModuleNgFactory:i0.NgModuleFactory<i1.MdTabsModule> = i0.ɵcmf(i1.MdTabsModule,
    ([] as any[]),(_l:any) => {
      return i0.ɵmod([i0.ɵmpd(512,i0.ComponentFactoryResolver,i0.ɵCodegenComponentFactoryResolver,
          [[8,([] as any[])],[3,i0.ComponentFactoryResolver],i0.NgModuleRef]),i0.ɵmpd(4608,
          i2.NgLocalization,i2.NgLocaleLocalization,[i0.LOCALE_ID]),i0.ɵmpd(6144,i3.DIR_DOCUMENT,
          (null as any),[i4.DOCUMENT]),i0.ɵmpd(4608,i3.Directionality,i3.Directionality,
          [[2,i3.DIR_DOCUMENT]]),i0.ɵmpd(4608,i5.Platform,i5.Platform,([] as any[])),
          i0.ɵmpd(5120,i6.ScrollDispatcher,i6.SCROLL_DISPATCHER_PROVIDER_FACTORY,[[3,
              i6.ScrollDispatcher],i0.NgZone,i5.Platform]),i0.ɵmpd(5120,i6.ViewportRuler,
              i6.VIEWPORT_RULER_PROVIDER_FACTORY,[[3,i6.ViewportRuler],i6.ScrollDispatcher]),
          i0.ɵmpd(4608,i7.MdMutationObserverFactory,i7.MdMutationObserverFactory,([] as any[])),
          i0.ɵmpd(512,i2.CommonModule,i2.CommonModule,([] as any[])),i0.ɵmpd(512,i8.CompatibilityModule,
              i8.CompatibilityModule,([] as any[])),i0.ɵmpd(512,i3.BidiModule,i3.BidiModule,
              ([] as any[])),i0.ɵmpd(256,i8.MATERIAL_SANITY_CHECKS,true,([] as any[])),
          i0.ɵmpd(512,i8.MdCommonModule,i8.MdCommonModule,[[2,i4.DOCUMENT],[2,i8.MATERIAL_SANITY_CHECKS]]),
          i0.ɵmpd(512,i9.PortalModule,i9.PortalModule,([] as any[])),i0.ɵmpd(512,i5.PlatformModule,
              i5.PlatformModule,([] as any[])),i0.ɵmpd(512,i6.ScrollDispatchModule,
              i6.ScrollDispatchModule,([] as any[])),i0.ɵmpd(512,i8.MdRippleModule,
              i8.MdRippleModule,([] as any[])),i0.ɵmpd(512,i7.ObserversModule,i7.ObserversModule,
              ([] as any[])),i0.ɵmpd(512,i1.MdTabsModule,i1.MdTabsModule,([] as any[]))]);
    });
const styles_MdTabGroup:any[] = ['.mat-tab-group{display:flex;flex-direction:column}.mat-tab-group.mat-tab-group-inverted-header{flex-direction:column-reverse}.mat-tab-label{height:48px;padding:0 24px;cursor:pointer;box-sizing:border-box;opacity:.6;min-width:160px;text-align:center;display:inline-flex;justify-content:center;align-items:center;white-space:nowrap;position:relative}.mat-tab-label:focus{outline:0;opacity:1}.mat-tab-label.mat-tab-disabled{cursor:default;pointer-events:none}@media (max-width:600px){.mat-tab-label{padding:0 12px}}@media (max-width:960px){.mat-tab-label{padding:0 12px}}.mat-tab-group[mat-stretch-tabs] .mat-tab-label,.mat-tab-group[md-stretch-tabs] .mat-tab-label{flex-basis:0;flex-grow:1}.mat-tab-body-wrapper{position:relative;overflow:hidden;display:flex;transition:height .5s cubic-bezier(.35,0,.25,1)}.mat-tab-body{top:0;left:0;right:0;bottom:0;position:absolute;display:block;overflow:hidden}.mat-tab-body.mat-tab-body-active{position:relative;overflow-x:hidden;overflow-y:auto;z-index:1;flex-grow:1}.mat-tab-group.mat-tab-group-dynamic-height .mat-tab-body.mat-tab-body-active{overflow-y:hidden}'];
export const RenderType_MdTabGroup:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_MdTabGroup,
    data:{}});
function View_MdTabGroup_3(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵand(0,(null as any),(null as any),0))],(null as any),
      (null as any));
}
function View_MdTabGroup_2(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),
      View_MdTabGroup_3)),i0.ɵdid(147456,(null as any),0,i9.PortalHostDirective,[i0.ComponentFactoryResolver,
      i0.ViewContainerRef],{portal:[0,'portal']},(null as any)),(_l()(),i0.ɵand(0,
      (null as any),(null as any),0))],(_ck,_v) => {
    const currVal_0:any = (<any>_v.parent).context.$implicit.templateLabel;
    _ck(_v,1,0,currVal_0);
  },(null as any));
}
function View_MdTabGroup_4(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵted((null as any),['','']))],(null as any),(_ck,_v) => {
    const currVal_0:any = (<any>_v.parent).context.$implicit.textLabel;
    _ck(_v,0,0,currVal_0);
  });
}
function View_MdTabGroup_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),6,'div',[['class',
      'mat-tab-label mat-ripple'],['mat-ripple',''],['matTabLabelWrapper',''],['role',
      'tab']],[[8,'id',0],[8,'tabIndex',0],[1,'aria-controls',0],[1,'aria-selected',
      0],[2,'mat-tab-label-active',(null as any)],[2,'mat-ripple-unbounded',(null as any)],
      [2,'mat-tab-disabled',(null as any)]],[[(null as any),'click']],(_v,en,$event) => {
    var ad:boolean = true;
    var _co:any = _v.component;
    if (('click' === en)) {
      const pd_0:any = ((<any>(i0.ɵnov((<any>_v.parent),2).focusIndex = (_co.selectedIndex = _v.context.index))) !== false);
      ad = (pd_0 && ad);
    }
    return ad;
  },(null as any),(null as any))),i0.ɵdid(671744,(null as any),0,i8.MdRipple,[i0.ElementRef,
      i0.NgZone,i6.ViewportRuler,i5.Platform,[2,i8.MD_RIPPLE_GLOBAL_OPTIONS]],{_matRippleDisabled:[0,
      '_matRippleDisabled']},(null as any)),i0.ɵdid(16384,[[2,4]],0,i1.MdTabLabelWrapper,
      [i0.ElementRef],{disabled:[0,'disabled']},(null as any)),(_l()(),i0.ɵand(16777216,
      (null as any),(null as any),1,(null as any),View_MdTabGroup_2)),i0.ɵdid(16384,
      (null as any),0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},
      (null as any)),(_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),
      View_MdTabGroup_4)),i0.ɵdid(16384,(null as any),0,i2.NgIf,[i0.ViewContainerRef,
      i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_7:any = _co.disableRipple;
    _ck(_v,1,0,currVal_7);
    const currVal_8:any = _v.context.$implicit.disabled;
    _ck(_v,2,0,currVal_8);
    const currVal_9:any = _v.context.$implicit.templateLabel;
    _ck(_v,4,0,currVal_9);
    const currVal_10:boolean = !_v.context.$implicit.templateLabel;
    _ck(_v,6,0,currVal_10);
  },(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co._getTabLabelId(_v.context.index);
    const currVal_1:any = ((_co.selectedIndex == _v.context.index)? 0: (0 - 1));
    const currVal_2:any = _co._getTabContentId(_v.context.index);
    const currVal_3:any = (_co.selectedIndex == _v.context.index);
    const currVal_4:any = (_co.selectedIndex == _v.context.index);
    const currVal_5:any = i0.ɵnov(_v,1).unbounded;
    const currVal_6:any = i0.ɵnov(_v,2).disabled;
    _ck(_v,0,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4,currVal_5,currVal_6);
  });
}
function View_MdTabGroup_5(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'mat-tab-body',
      [['class','mat-tab-body'],['role','tabpanel']],[[8,'id',0],[1,'aria-labelledby',
          0],[2,'mat-tab-body-active',(null as any)]],[[(null as any),'onCentered'],
          [(null as any),'onCentering']],(_v,en,$event) => {
        var ad:boolean = true;
        var _co:any = _v.component;
        if (('onCentered' === en)) {
          const pd_0:any = ((<any>_co._removeTabBodyWrapperHeight()) !== false);
          ad = (pd_0 && ad);
        }
        if (('onCentering' === en)) {
          const pd_1:any = ((<any>_co._setTabBodyWrapperHeight($event)) !== false);
          ad = (pd_1 && ad);
        }
        return ad;
      },View_MdTabBody_0,RenderType_MdTabBody)),i0.ɵdid(8503296,(null as any),0,i1.MdTabBody,
      [i0.ElementRef,[2,i3.Directionality]],{_content:[0,'_content'],position:[1,'position'],
          origin:[2,'origin']},{onCentering:'onCentering',onCentered:'onCentered'})],
      (_ck,_v) => {
        const currVal_3:any = _v.context.$implicit.content;
        const currVal_4:any = _v.context.$implicit.position;
        const currVal_5:any = _v.context.$implicit.origin;
        _ck(_v,1,0,currVal_3,currVal_4,currVal_5);
      },(_ck,_v) => {
        var _co:any = _v.component;
        const currVal_0:any = _co._getTabContentId(_v.context.index);
        const currVal_1:any = _co._getTabLabelId(_v.context.index);
        const currVal_2:any = (_co.selectedIndex == _v.context.index);
        _ck(_v,0,0,currVal_0,currVal_1,currVal_2);
      });
}
export function View_MdTabGroup_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(2,[i0.ɵqud(402653184,1,{_tabBodyWrapper:0}),(_l()(),i0.ɵeld(0,(null as any),
      (null as any),5,'mat-tab-header',[['class','mat-tab-header']],[[2,'mat-tab-header-pagination-controls-enabled',
          (null as any)],[2,'mat-tab-header-rtl',(null as any)]],[[(null as any),'indexFocused'],
          [(null as any),'selectFocusedIndex']],(_v,en,$event) => {
        var ad:boolean = true;
        var _co:i1.MdTabGroup = _v.component;
        if (('indexFocused' === en)) {
          const pd_0:any = ((<any>_co._focusChanged($event)) !== false);
          ad = (pd_0 && ad);
        }
        if (('selectFocusedIndex' === en)) {
          const pd_1:any = ((<any>(_co.selectedIndex = $event)) !== false);
          ad = (pd_1 && ad);
        }
        return ad;
      },View_MdTabHeader_0,RenderType_MdTabHeader)),i0.ɵdid(3325952,[['tabHeader',
      4]],1,i1.MdTabHeader,[i0.ElementRef,i0.Renderer2,i0.ChangeDetectorRef,[2,i3.Directionality]],
      {disableRipple:[0,'disableRipple'],selectedIndex:[1,'selectedIndex']},{selectFocusedIndex:'selectFocusedIndex',
          indexFocused:'indexFocused'}),i0.ɵqud(603979776,2,{_labelWrappers:1}),i0.ɵprd(8448,
      (null as any),i8.MATERIAL_COMPATIBILITY_MODE,true,([] as any[])),(_l()(),i0.ɵand(16777216,
      (null as any),0,1,(null as any),View_MdTabGroup_1)),i0.ɵdid(802816,(null as any),
      0,i2.NgForOf,[i0.ViewContainerRef,i0.TemplateRef,i0.IterableDiffers],{ngForOf:[0,
          'ngForOf']},(null as any)),(_l()(),i0.ɵeld(0,[[1,0],['tabBodyWrapper',1]],
      (null as any),2,'div',[['class','mat-tab-body-wrapper']],(null as any),(null as any),
      (null as any),(null as any),(null as any))),(_l()(),i0.ɵand(16777216,(null as any),
      (null as any),1,(null as any),View_MdTabGroup_5)),i0.ɵdid(802816,(null as any),
      0,i2.NgForOf,[i0.ViewContainerRef,i0.TemplateRef,i0.IterableDiffers],{ngForOf:[0,
          'ngForOf']},(null as any))],(_ck,_v) => {
    var _co:i1.MdTabGroup = _v.component;
    const currVal_2:any = _co.disableRipple;
    const currVal_3:any = _co.selectedIndex;
    _ck(_v,2,0,currVal_2,currVal_3);
    const currVal_4:any = _co._tabs;
    _ck(_v,6,0,currVal_4);
    const currVal_5:any = _co._tabs;
    _ck(_v,9,0,currVal_5);
  },(_ck,_v) => {
    const currVal_0:any = i0.ɵnov(_v,2)._showPaginationControls;
    const currVal_1:any = (i0.ɵnov(_v,2)._getLayoutDirection() == 'rtl');
    _ck(_v,1,0,currVal_0,currVal_1);
  });
}
export function View_MdTabGroup_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),3,'md-tab-group',
      [['class','mat-tab-group']],[[2,'mat-tab-group-dynamic-height',(null as any)],
          [2,'mat-tab-group-inverted-header',(null as any)]],(null as any),(null as any),
      View_MdTabGroup_0,RenderType_MdTabGroup)),i0.ɵdid(11714560,(null as any),1,i1.MdTabGroup,
      [i0.Renderer2,i0.ElementRef,i0.ChangeDetectorRef],(null as any),(null as any)),
      i0.ɵqud(603979776,1,{_tabs:1}),i0.ɵprd(8448,(null as any),i8.MATERIAL_COMPATIBILITY_MODE,
          true,([] as any[]))],(null as any),(_ck,_v) => {
    const currVal_0:any = i0.ɵnov(_v,1).dynamicHeight;
    const currVal_1:any = (i0.ɵnov(_v,1).headerPosition === 'below');
    _ck(_v,0,0,currVal_0,currVal_1);
  });
}
export const MdTabGroupNgFactory:i0.ComponentFactory<i1.MdTabGroup> = i0.ɵccf('md-tab-group, mat-tab-group',
    i1.MdTabGroup,View_MdTabGroup_Host_0,{color:'color',disableRipple:'disableRipple',
        dynamicHeight:'dynamicHeight',_dynamicHeightDeprecated:'md-dynamic-height',
        selectedIndex:'selectedIndex',headerPosition:'headerPosition',backgroundColor:'backgroundColor'},
    {selectedIndexChange:'selectedIndexChange',focusChange:'focusChange',selectChange:'selectChange'},
    ([] as any[]));
const styles_MdTab:any[] = ([] as any[]);
export const RenderType_MdTab:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_MdTab,
    data:{}});
function View_MdTab_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[i0.ɵncd((null as any),0),(_l()(),i0.ɵand(0,(null as any),(null as any),
      0))],(null as any),(null as any));
}
export function View_MdTab_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(2,[i0.ɵqud(402653184,1,{_content:0}),(_l()(),i0.ɵand(0,[[1,2]],(null as any),
      0,(null as any),View_MdTab_1))],(null as any),(null as any));
}
export function View_MdTab_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(16777216,(null as any),(null as any),2,'md-tab',
      ([] as any[]),(null as any),(null as any),(null as any),View_MdTab_0,RenderType_MdTab)),
      i0.ɵdid(770048,(null as any),1,i1.MdTab,[i0.ViewContainerRef],(null as any),
          (null as any)),i0.ɵqud(335544320,1,{templateLabel:0})],(_ck,_v) => {
    _ck(_v,1,0);
  },(null as any));
}
export const MdTabNgFactory:i0.ComponentFactory<i1.MdTab> = i0.ɵccf('md-tab, mat-tab',
    i1.MdTab,View_MdTab_Host_0,{disabled:'disabled',textLabel:'label'},{},['*']);
const styles_MdTabNav:any[] = ['.mat-tab-nav-bar{overflow:hidden;position:relative;flex-shrink:0}.mat-tab-links{position:relative}.mat-tab-link{height:48px;padding:0 24px;cursor:pointer;box-sizing:border-box;opacity:.6;min-width:160px;text-align:center;display:inline-flex;justify-content:center;align-items:center;white-space:nowrap;vertical-align:top;text-decoration:none;position:relative;overflow:hidden}.mat-tab-link:focus{outline:0;opacity:1}.mat-tab-link.mat-tab-disabled{cursor:default;pointer-events:none}@media (max-width:600px){.mat-tab-link{min-width:72px}}.mat-ink-bar{position:absolute;bottom:0;height:2px;transition:.5s cubic-bezier(.35,0,.25,1)}.mat-tab-group-inverted-header .mat-ink-bar{bottom:auto;top:0}'];
export const RenderType_MdTabNav:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_MdTabNav,
    data:{}});
export function View_MdTabNav_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(2,[i0.ɵqud(402653184,1,{_inkBar:0}),(_l()(),i0.ɵeld(0,(null as any),
      (null as any),4,'div',[['class','mat-tab-links']],(null as any),[[(null as any),
          'cdkObserveContent']],(_v,en,$event) => {
        var ad:boolean = true;
        var _co:i1.MdTabNav = _v.component;
        if (('cdkObserveContent' === en)) {
          const pd_0:any = ((<any>_co._alignInkBar()) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },(null as any),(null as any))),i0.ɵdid(1196032,(null as any),0,i7.ObserveContent,
      [i7.MdMutationObserverFactory,i0.ElementRef,i0.NgZone],(null as any),{event:'cdkObserveContent'}),
      i0.ɵncd((null as any),0),(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'mat-ink-bar',
          [['class','mat-ink-bar']],(null as any),(null as any),(null as any),(null as any),
          (null as any))),i0.ɵdid(16384,[[1,4]],0,i1.MdInkBar,[i0.Renderer2,i0.ElementRef,
          i0.NgZone],(null as any),(null as any))],(null as any),(null as any));
}
export function View_MdTabNav_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),3,'div',[['class',
      'mat-tab-nav-bar'],['md-tab-nav-bar','']],(null as any),(null as any),(null as any),
      View_MdTabNav_0,RenderType_MdTabNav)),i0.ɵdid(3325952,(null as any),1,i1.MdTabNav,
      [i0.Renderer2,i0.ElementRef,[2,i3.Directionality],i0.NgZone,i0.ChangeDetectorRef],
      (null as any),(null as any)),i0.ɵqud(603979776,1,{_tabLinks:1}),i0.ɵprd(8448,
      (null as any),i8.MATERIAL_COMPATIBILITY_MODE,true,([] as any[]))],(null as any),
      (null as any));
}
export const MdTabNavNgFactory:i0.ComponentFactory<i1.MdTabNav> = i0.ɵccf('[md-tab-nav-bar], [mat-tab-nav-bar]',
    i1.MdTabNav,View_MdTabNav_Host_0,{color:'color',disableRipple:'disableRipple',
        backgroundColor:'backgroundColor'},{},['*']);
const styles_MdTabBody:any[] = ['.mat-tab-body-content{height:100%;overflow:auto}.mat-tab-group-dynamic-height .mat-tab-body-content{overflow:hidden}'];
export const RenderType_MdTabBody:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_MdTabBody,
    data:{'animation':[{type:7,name:'translateTab',definitions:[{type:0,name:'void',
        styles:{type:6,styles:{transform:'translate3d(0%, 0, 0)'},offset:(null as any)},
        options:(undefined as any)},{type:0,name:'left',styles:{type:6,styles:{transform:'translate3d(-100%, 0, 0)'},
        offset:(null as any)},options:(undefined as any)},{type:0,name:'left-origin-center',
        styles:{type:6,styles:{transform:'translate3d(0%, 0, 0)'},offset:(null as any)},
        options:(undefined as any)},{type:0,name:'right-origin-center',styles:{type:6,
        styles:{transform:'translate3d(0%, 0, 0)'},offset:(null as any)},options:(undefined as any)},
        {type:0,name:'center',styles:{type:6,styles:{transform:'translate3d(0%, 0, 0)'},
            offset:(null as any)},options:(undefined as any)},{type:0,name:'right',
            styles:{type:6,styles:{transform:'translate3d(100%, 0, 0)'},offset:(null as any)},
            options:(undefined as any)},{type:1,expr:'* => left, * => right, left => center, right => center',
            animation:{type:4,styles:(null as any),timings:'500ms cubic-bezier(0.35, 0, 0.25, 1)'},
            options:(null as any)},{type:1,expr:'void => left-origin-center',animation:[{type:6,
            styles:{transform:'translate3d(-100%, 0, 0)'},offset:(null as any)},{type:4,
            styles:(null as any),timings:'500ms cubic-bezier(0.35, 0, 0.25, 1)'}],
            options:(null as any)},{type:1,expr:'void => right-origin-center',animation:[{type:6,
            styles:{transform:'translate3d(100%, 0, 0)'},offset:(null as any)},{type:4,
            styles:(null as any),timings:'500ms cubic-bezier(0.35, 0, 0.25, 1)'}],
            options:(null as any)}],options:{}}]}});
function View_MdTabBody_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵand(0,(null as any),(null as any),0))],(null as any),
      (null as any));
}
export function View_MdTabBody_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(2,[i0.ɵqud(402653184,1,{_portalHost:0}),(_l()(),i0.ɵeld(0,[['content',
      1]],(null as any),2,'div',[['class','mat-tab-body-content']],[[24,'@translateTab',
      0]],[[(null as any),'@translateTab.start'],[(null as any),'@translateTab.done']],
      (_v,en,$event) => {
        var ad:boolean = true;
        var _co:i1.MdTabBody = _v.component;
        if (('@translateTab.start' === en)) {
          const pd_0:any = ((<any>_co._onTranslateTabStarted($event)) !== false);
          ad = (pd_0 && ad);
        }
        if (('@translateTab.done' === en)) {
          const pd_1:any = ((<any>_co._onTranslateTabComplete($event)) !== false);
          ad = (pd_1 && ad);
        }
        return ad;
      },(null as any),(null as any))),(_l()(),i0.ɵand(16777216,(null as any),(null as any),
      1,(null as any),View_MdTabBody_1)),i0.ɵdid(147456,[[1,4]],0,i9.PortalHostDirective,
      [i0.ComponentFactoryResolver,i0.ViewContainerRef],{portal:[0,'portal']},(null as any))],
      (_ck,_v) => {
        const currVal_1:any = '';
        _ck(_v,3,0,currVal_1);
      },(_ck,_v) => {
        var _co:i1.MdTabBody = _v.component;
        const currVal_0:any = _co._position;
        _ck(_v,1,0,currVal_0);
      });
}
export function View_MdTabBody_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'md-tab-body',
      [['class','mat-tab-body']],(null as any),(null as any),(null as any),View_MdTabBody_0,
      RenderType_MdTabBody)),i0.ɵdid(8503296,(null as any),0,i1.MdTabBody,[i0.ElementRef,
      [2,i3.Directionality]],(null as any),(null as any))],(_ck,_v) => {
    _ck(_v,1,0);
  },(null as any));
}
export const MdTabBodyNgFactory:i0.ComponentFactory<i1.MdTabBody> = i0.ɵccf('md-tab-body, mat-tab-body',
    i1.MdTabBody,View_MdTabBody_Host_0,{_content:'content',position:'position',origin:'origin'},
    {onCentering:'onCentering',onCentered:'onCentered'},([] as any[]));
const styles_MdTabHeader:any[] = ['.mat-tab-header{display:flex;overflow:hidden;position:relative;flex-shrink:0}.mat-tab-label{height:48px;padding:0 24px;cursor:pointer;box-sizing:border-box;opacity:.6;min-width:160px;text-align:center;display:inline-flex;justify-content:center;align-items:center;white-space:nowrap;position:relative}.mat-tab-label:focus{outline:0;opacity:1}.mat-tab-label.mat-tab-disabled{cursor:default;pointer-events:none}@media (max-width:600px){.mat-tab-label{min-width:72px}}.mat-ink-bar{position:absolute;bottom:0;height:2px;transition:.5s cubic-bezier(.35,0,.25,1)}.mat-tab-group-inverted-header .mat-ink-bar{bottom:auto;top:0}.mat-tab-header-pagination{position:relative;display:none;justify-content:center;align-items:center;min-width:32px;cursor:pointer;z-index:2}.mat-tab-header-pagination-controls-enabled .mat-tab-header-pagination{display:flex}.mat-tab-header-pagination-before,.mat-tab-header-rtl .mat-tab-header-pagination-after{padding-left:4px}.mat-tab-header-pagination-before .mat-tab-header-pagination-chevron,.mat-tab-header-rtl .mat-tab-header-pagination-after .mat-tab-header-pagination-chevron{transform:rotate(-135deg)}.mat-tab-header-pagination-after,.mat-tab-header-rtl .mat-tab-header-pagination-before{padding-right:4px}.mat-tab-header-pagination-after .mat-tab-header-pagination-chevron,.mat-tab-header-rtl .mat-tab-header-pagination-before .mat-tab-header-pagination-chevron{transform:rotate(45deg)}.mat-tab-header-pagination-chevron{border-style:solid;border-width:2px 2px 0 0;content:\'\';height:8px;width:8px}.mat-tab-header-pagination-disabled{box-shadow:none;cursor:default}.mat-tab-label-container{display:flex;flex-grow:1;overflow:hidden;z-index:1}.mat-tab-list{flex-grow:1;position:relative;transition:transform .5s cubic-bezier(.35,0,.25,1)}.mat-tab-labels{display:flex}'];
export const RenderType_MdTabHeader:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_MdTabHeader,
    data:{}});
export function View_MdTabHeader_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(2,[i0.ɵqud(402653184,1,{_inkBar:0}),i0.ɵqud(402653184,2,{_tabListContainer:0}),
      i0.ɵqud(402653184,3,{_tabList:0}),(_l()(),i0.ɵeld(0,(null as any),(null as any),
          2,'div',[['aria-hidden','true'],['class','mat-tab-header-pagination mat-tab-header-pagination-before mat-elevation-z4 mat-ripple'],
              ['mat-ripple','']],[[2,'mat-tab-header-pagination-disabled',(null as any)],
              [2,'mat-ripple-unbounded',(null as any)]],[[(null as any),'click']],
          (_v,en,$event) => {
            var ad:boolean = true;
            var _co:i1.MdTabHeader = _v.component;
            if (('click' === en)) {
              const pd_0:any = ((<any>_co._scrollHeader('before')) !== false);
              ad = (pd_0 && ad);
            }
            return ad;
          },(null as any),(null as any))),i0.ɵdid(671744,(null as any),0,i8.MdRipple,
          [i0.ElementRef,i0.NgZone,i6.ViewportRuler,i5.Platform,[2,i8.MD_RIPPLE_GLOBAL_OPTIONS]],
          {_matRippleDisabled:[0,'_matRippleDisabled']},(null as any)),(_l()(),i0.ɵeld(0,
          (null as any),(null as any),0,'div',[['class','mat-tab-header-pagination-chevron']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵeld(0,[[2,0],['tabListContainer',1]],(null as any),6,'div',[['class',
          'mat-tab-label-container']],(null as any),[[(null as any),'keydown']],(_v,
          en,$event) => {
        var ad:boolean = true;
        var _co:i1.MdTabHeader = _v.component;
        if (('keydown' === en)) {
          const pd_0:any = ((<any>_co._handleKeydown($event)) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },(null as any),(null as any))),(_l()(),i0.ɵeld(0,[[3,0],['tabList',1]],(null as any),
          5,'div',[['class','mat-tab-list'],['role','tablist']],(null as any),[[(null as any),
              'cdkObserveContent']],(_v,en,$event) => {
            var ad:boolean = true;
            var _co:i1.MdTabHeader = _v.component;
            if (('cdkObserveContent' === en)) {
              const pd_0:any = ((<any>_co._onContentChanges()) !== false);
              ad = (pd_0 && ad);
            }
            return ad;
          },(null as any),(null as any))),i0.ɵdid(1196032,(null as any),0,i7.ObserveContent,
          [i7.MdMutationObserverFactory,i0.ElementRef,i0.NgZone],(null as any),{event:'cdkObserveContent'}),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),1,'div',[['class','mat-tab-labels']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      i0.ɵncd((null as any),0),(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'mat-ink-bar',
          [['class','mat-ink-bar']],(null as any),(null as any),(null as any),(null as any),
          (null as any))),i0.ɵdid(16384,[[1,4]],0,i1.MdInkBar,[i0.Renderer2,i0.ElementRef,
          i0.NgZone],(null as any),(null as any)),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),2,'div',[['aria-hidden','true'],['class','mat-tab-header-pagination mat-tab-header-pagination-after mat-elevation-z4 mat-ripple'],
              ['mat-ripple','']],[[2,'mat-tab-header-pagination-disabled',(null as any)],
              [2,'mat-ripple-unbounded',(null as any)]],[[(null as any),'click']],
          (_v,en,$event) => {
            var ad:boolean = true;
            var _co:i1.MdTabHeader = _v.component;
            if (('click' === en)) {
              const pd_0:any = ((<any>_co._scrollHeader('after')) !== false);
              ad = (pd_0 && ad);
            }
            return ad;
          },(null as any),(null as any))),i0.ɵdid(671744,(null as any),0,i8.MdRipple,
          [i0.ElementRef,i0.NgZone,i6.ViewportRuler,i5.Platform,[2,i8.MD_RIPPLE_GLOBAL_OPTIONS]],
          {_matRippleDisabled:[0,'_matRippleDisabled']},(null as any)),(_l()(),i0.ɵeld(0,
          (null as any),(null as any),0,'div',[['class','mat-tab-header-pagination-chevron']],
          (null as any),(null as any),(null as any),(null as any),(null as any)))],
      (_ck,_v) => {
        var _co:i1.MdTabHeader = _v.component;
        const currVal_2:any = (_co._disableScrollBefore || _co.disableRipple);
        _ck(_v,4,0,currVal_2);
        const currVal_5:any = (_co._disableScrollAfter || _co.disableRipple);
        _ck(_v,14,0,currVal_5);
      },(_ck,_v) => {
        var _co:i1.MdTabHeader = _v.component;
        const currVal_0:any = _co._disableScrollBefore;
        const currVal_1:any = i0.ɵnov(_v,4).unbounded;
        _ck(_v,3,0,currVal_0,currVal_1);
        const currVal_3:any = _co._disableScrollAfter;
        const currVal_4:any = i0.ɵnov(_v,14).unbounded;
        _ck(_v,13,0,currVal_3,currVal_4);
      });
}
export function View_MdTabHeader_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),3,'md-tab-header',
      [['class','mat-tab-header']],[[2,'mat-tab-header-pagination-controls-enabled',
          (null as any)],[2,'mat-tab-header-rtl',(null as any)]],(null as any),(null as any),
      View_MdTabHeader_0,RenderType_MdTabHeader)),i0.ɵdid(3325952,(null as any),1,
      i1.MdTabHeader,[i0.ElementRef,i0.Renderer2,i0.ChangeDetectorRef,[2,i3.Directionality]],
      (null as any),(null as any)),i0.ɵqud(603979776,1,{_labelWrappers:1}),i0.ɵprd(8448,
      (null as any),i8.MATERIAL_COMPATIBILITY_MODE,true,([] as any[]))],(null as any),
      (_ck,_v) => {
        const currVal_0:any = i0.ɵnov(_v,1)._showPaginationControls;
        const currVal_1:any = (i0.ɵnov(_v,1)._getLayoutDirection() == 'rtl');
        _ck(_v,0,0,currVal_0,currVal_1);
      });
}
export const MdTabHeaderNgFactory:i0.ComponentFactory<i1.MdTabHeader> = i0.ɵccf('md-tab-header, mat-tab-header',
    i1.MdTabHeader,View_MdTabHeader_Host_0,{disableRipple:'disableRipple',selectedIndex:'selectedIndex'},
    {selectFocusedIndex:'selectFocusedIndex',indexFocused:'indexFocused'},['*']);
