// @angular/material/progress-spinner/typings/index.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from '@angular/material/progress-spinner';
import * as i2 from '@angular/cdk/bidi';
import * as i3 from '@angular/platform-browser';
import * as i4 from '@angular/material/core';
export const MdProgressSpinnerModuleNgFactory:i0.NgModuleFactory<i1.MdProgressSpinnerModule> = i0.ɵcmf(i1.MdProgressSpinnerModule,
    ([] as any[]),(_l:any) => {
      return i0.ɵmod([i0.ɵmpd(512,i0.ComponentFactoryResolver,i0.ɵCodegenComponentFactoryResolver,
          [[8,([] as any[])],[3,i0.ComponentFactoryResolver],i0.NgModuleRef]),i0.ɵmpd(6144,
          i2.DIR_DOCUMENT,(null as any),[i3.DOCUMENT]),i0.ɵmpd(4608,i2.Directionality,
          i2.Directionality,[[2,i2.DIR_DOCUMENT]]),i0.ɵmpd(512,i4.CompatibilityModule,
          i4.CompatibilityModule,([] as any[])),i0.ɵmpd(512,i2.BidiModule,i2.BidiModule,
          ([] as any[])),i0.ɵmpd(256,i4.MATERIAL_SANITY_CHECKS,true,([] as any[])),
          i0.ɵmpd(512,i4.MdCommonModule,i4.MdCommonModule,[[2,i3.DOCUMENT],[2,i4.MATERIAL_SANITY_CHECKS]]),
          i0.ɵmpd(512,i1.MdProgressSpinnerModule,i1.MdProgressSpinnerModule,([] as any[]))]);
    });
const styles_MdProgressSpinner:any[] = ['.mat-progress-spinner{display:block;height:100px;width:100px;overflow:hidden}.mat-progress-spinner svg{height:100%;width:100%;transform-origin:center}.mat-progress-spinner path{fill:transparent;transition:stroke .3s cubic-bezier(.35,0,.25,1)}.mat-progress-spinner[mode=indeterminate] svg{animation-duration:5.25s,2.887s;animation-name:mat-progress-spinner-sporadic-rotate,mat-progress-spinner-linear-rotate;animation-timing-function:cubic-bezier(.35,0,.25,1),linear;animation-iteration-count:infinite;transition:none}@keyframes mat-progress-spinner-linear-rotate{0%{transform:rotate(0)}100%{transform:rotate(360deg)}}@keyframes mat-progress-spinner-sporadic-rotate{12.5%{transform:rotate(135deg)}25%{transform:rotate(270deg)}37.5%{transform:rotate(405deg)}50%{transform:rotate(540deg)}62.5%{transform:rotate(675deg)}75%{transform:rotate(810deg)}87.5%{transform:rotate(945deg)}100%{transform:rotate(1080deg)}}'];
export const RenderType_MdProgressSpinner:i0.RendererType2 = i0.ɵcrt({encapsulation:2,
    styles:styles_MdProgressSpinner,data:{}});
export function View_MdProgressSpinner_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(2,[i0.ɵqud(402653184,1,{_path:0}),(_l()(),i0.ɵeld(0,(null as any),
      (null as any),1,':svg:svg',[['focusable','false'],['preserveAspectRatio','xMidYMid meet'],
          ['viewBox','0 0 100 100']],(null as any),(null as any),(null as any),(null as any),
      (null as any))),(_l()(),i0.ɵeld(0,[[1,0],['path',1]],(null as any),0,':svg:path',
      ([] as any[]),[[4,'strokeWidth',(null as any)]],(null as any),(null as any),
      (null as any),(null as any)))],(null as any),(_ck,_v) => {
    var _co:i1.MdProgressSpinner = _v.component;
    const currVal_0:any = _co.strokeWidth;
    _ck(_v,2,0,currVal_0);
  });
}
export function View_MdProgressSpinner_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'md-progress-spinner',
      [['class','mat-progress-spinner'],['role','progressbar']],[[1,'aria-valuemin',
          0],[1,'aria-valuemax',0],[1,'aria-valuenow',0],[1,'mode',0]],(null as any),
      (null as any),View_MdProgressSpinner_0,RenderType_MdProgressSpinner)),i0.ɵdid(180224,
      (null as any),0,i1.MdProgressSpinner,[i0.Renderer2,i0.ElementRef,i0.NgZone],
      (null as any),(null as any))],(null as any),(_ck,_v) => {
    const currVal_0:any = i0.ɵnov(_v,1)._ariaValueMin;
    const currVal_1:any = i0.ɵnov(_v,1)._ariaValueMax;
    const currVal_2:any = i0.ɵnov(_v,1).value;
    const currVal_3:any = i0.ɵnov(_v,1).mode;
    _ck(_v,0,0,currVal_0,currVal_1,currVal_2,currVal_3);
  });
}
export const MdProgressSpinnerNgFactory:i0.ComponentFactory<i1.MdProgressSpinner> = i0.ɵccf('md-progress-spinner, mat-progress-spinner',
    i1.MdProgressSpinner,View_MdProgressSpinner_Host_0,{color:'color',strokeWidth:'strokeWidth',
        value:'value',mode:'mode'},{},([] as any[]));
const styles_MdSpinner:any[] = ['.mat-progress-spinner{display:block;height:100px;width:100px;overflow:hidden}.mat-progress-spinner svg{height:100%;width:100%;transform-origin:center}.mat-progress-spinner path{fill:transparent;transition:stroke .3s cubic-bezier(.35,0,.25,1)}.mat-progress-spinner[mode=indeterminate] svg{animation-duration:5.25s,2.887s;animation-name:mat-progress-spinner-sporadic-rotate,mat-progress-spinner-linear-rotate;animation-timing-function:cubic-bezier(.35,0,.25,1),linear;animation-iteration-count:infinite;transition:none}@keyframes mat-progress-spinner-linear-rotate{0%{transform:rotate(0)}100%{transform:rotate(360deg)}}@keyframes mat-progress-spinner-sporadic-rotate{12.5%{transform:rotate(135deg)}25%{transform:rotate(270deg)}37.5%{transform:rotate(405deg)}50%{transform:rotate(540deg)}62.5%{transform:rotate(675deg)}75%{transform:rotate(810deg)}87.5%{transform:rotate(945deg)}100%{transform:rotate(1080deg)}}'];
export const RenderType_MdSpinner:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_MdSpinner,
    data:{}});
export function View_MdSpinner_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(2,[i0.ɵqud(402653184,1,{_path:0}),(_l()(),i0.ɵeld(0,(null as any),
      (null as any),1,':svg:svg',[['focusable','false'],['preserveAspectRatio','xMidYMid meet'],
          ['viewBox','0 0 100 100']],(null as any),(null as any),(null as any),(null as any),
      (null as any))),(_l()(),i0.ɵeld(0,[[1,0],['path',1]],(null as any),0,':svg:path',
      ([] as any[]),[[4,'strokeWidth',(null as any)]],(null as any),(null as any),
      (null as any),(null as any)))],(null as any),(_ck,_v) => {
    var _co:i1.MdSpinner = _v.component;
    const currVal_0:any = _co.strokeWidth;
    _ck(_v,2,0,currVal_0);
  });
}
export function View_MdSpinner_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'md-spinner',[['class',
      'mat-spinner mat-progress-spinner'],['mode','indeterminate'],['role','progressbar']],
      (null as any),(null as any),(null as any),View_MdSpinner_0,RenderType_MdSpinner)),
      i0.ɵdid(180224,(null as any),0,i1.MdSpinner,[i0.ElementRef,i0.NgZone,i0.Renderer2],
          (null as any),(null as any))],(null as any),(null as any));
}
export const MdSpinnerNgFactory:i0.ComponentFactory<i1.MdSpinner> = i0.ɵccf('md-spinner, mat-spinner',
    i1.MdSpinner,View_MdSpinner_Host_0,{color:'color',strokeWidth:'strokeWidth',value:'value',
        mode:'mode'},{},([] as any[]));
