// @angular/material/select/typings/index.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from '@angular/material/select';
import * as i2 from '@angular/common';
import * as i3 from '@angular/cdk/platform';
import * as i4 from '@angular/cdk/scrolling';
import * as i5 from '@angular/cdk/overlay';
import * as i6 from '@angular/cdk/bidi';
import * as i7 from '@angular/platform-browser';
import * as i8 from '@angular/cdk/portal';
import * as i9 from '@angular/material/core';
import * as i10 from '@angular/forms';
export const MdSelectModuleNgFactory:i0.NgModuleFactory<i1.MdSelectModule> = i0.ɵcmf(i1.MdSelectModule,
    ([] as any[]),(_l:any) => {
      return i0.ɵmod([i0.ɵmpd(512,i0.ComponentFactoryResolver,i0.ɵCodegenComponentFactoryResolver,
          [[8,([] as any[])],[3,i0.ComponentFactoryResolver],i0.NgModuleRef]),i0.ɵmpd(4608,
          i2.NgLocalization,i2.NgLocaleLocalization,[i0.LOCALE_ID]),i0.ɵmpd(4608,i3.Platform,
          i3.Platform,([] as any[])),i0.ɵmpd(5120,i4.ScrollDispatcher,i4.SCROLL_DISPATCHER_PROVIDER_FACTORY,
          [[3,i4.ScrollDispatcher],i0.NgZone,i3.Platform]),i0.ɵmpd(5120,i4.ViewportRuler,
          i4.VIEWPORT_RULER_PROVIDER_FACTORY,[[3,i4.ViewportRuler],i4.ScrollDispatcher]),
          i0.ɵmpd(4608,i5.ScrollStrategyOptions,i5.ScrollStrategyOptions,[i4.ScrollDispatcher,
              i4.ViewportRuler]),i0.ɵmpd(5120,i5.OverlayContainer,i5.ɵa,[[3,i5.OverlayContainer]]),
          i0.ɵmpd(4608,i5.ɵf,i5.ɵf,[i4.ViewportRuler]),i0.ɵmpd(4608,i5.Overlay,i5.Overlay,
              [i5.ScrollStrategyOptions,i5.OverlayContainer,i0.ComponentFactoryResolver,
                  i5.ɵf,i0.ApplicationRef,i0.Injector,i0.NgZone]),i0.ɵmpd(5120,i5.ɵc,
              i5.ɵd,[i5.Overlay]),i0.ɵmpd(6144,i6.DIR_DOCUMENT,(null as any),[i7.DOCUMENT]),
          i0.ɵmpd(4608,i6.Directionality,i6.Directionality,[[2,i6.DIR_DOCUMENT]]),
          i0.ɵmpd(5120,i1.MD_SELECT_SCROLL_STRATEGY,i1.MAT_SELECT_SCROLL_STRATEGY_PROVIDER_FACTORY,
              [i5.Overlay]),i0.ɵmpd(512,i2.CommonModule,i2.CommonModule,([] as any[])),
          i0.ɵmpd(512,i8.PortalModule,i8.PortalModule,([] as any[])),i0.ɵmpd(512,i3.PlatformModule,
              i3.PlatformModule,([] as any[])),i0.ɵmpd(512,i4.ScrollDispatchModule,
              i4.ScrollDispatchModule,([] as any[])),i0.ɵmpd(512,i5.OverlayModule,
              i5.OverlayModule,([] as any[])),i0.ɵmpd(512,i9.CompatibilityModule,i9.CompatibilityModule,
              ([] as any[])),i0.ɵmpd(512,i6.BidiModule,i6.BidiModule,([] as any[])),
          i0.ɵmpd(256,i9.MATERIAL_SANITY_CHECKS,true,([] as any[])),i0.ɵmpd(512,i9.MdCommonModule,
              i9.MdCommonModule,[[2,i7.DOCUMENT],[2,i9.MATERIAL_SANITY_CHECKS]]),i0.ɵmpd(512,
              i9.MdRippleModule,i9.MdRippleModule,([] as any[])),i0.ɵmpd(512,i9.MdPseudoCheckboxModule,
              i9.MdPseudoCheckboxModule,([] as any[])),i0.ɵmpd(512,i9.MdOptionModule,
              i9.MdOptionModule,([] as any[])),i0.ɵmpd(512,i1.MdSelectModule,i1.MdSelectModule,
              ([] as any[]))]);
    });
const styles_MdSelect:any[] = ['.mat-select{display:inline-block;outline:0}.mat-select-trigger{display:flex;align-items:center;height:30px;min-width:112px;cursor:pointer;position:relative;box-sizing:border-box}.mat-select-disabled .mat-select-trigger{-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;cursor:default}.mat-select-underline{position:absolute;bottom:0;left:0;right:0;height:1px}.mat-select:focus .mat-select-underline{height:2px}.mat-select-disabled .mat-select-underline{background-color:transparent;background-position:0 bottom}.mat-select-placeholder{position:relative;padding:0 2px;transform-origin:left top;flex-grow:1}.mat-select-placeholder.mat-floating-placeholder{top:-22px;left:-2px;text-align:left;transform:scale(.75)}[dir=rtl] .mat-select-placeholder{transform-origin:right top}[dir=rtl] .mat-select-placeholder.mat-floating-placeholder{left:2px;text-align:right}.mat-select-required .mat-select-placeholder::after{content:\' *\'}.mat-select-value{position:absolute;max-width:calc(100% - 18px);flex-grow:1;top:0;left:0;bottom:0;display:flex;align-items:center}[dir=rtl] .mat-select-value{left:auto;right:0}.mat-select-value-text{white-space:nowrap;overflow:hidden;text-overflow:ellipsis;line-height:30px}.mat-select-arrow{width:0;height:0;border-left:5px solid transparent;border-right:5px solid transparent;border-top:5px solid;margin:0 4px}.mat-select-panel{min-width:112px;max-width:280px;overflow:auto;-webkit-overflow-scrolling:touch;padding-top:0;padding-bottom:0;max-height:256px;min-width:100%}.mat-select-panel:not([class*=mat-elevation-z]){box-shadow:0 5px 5px -3px rgba(0,0,0,.2),0 8px 10px 1px rgba(0,0,0,.14),0 3px 14px 2px rgba(0,0,0,.12)}@media screen and (-ms-high-contrast:active){.mat-select-panel{outline:solid 1px}}'];
export const RenderType_MdSelect:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_MdSelect,
    data:{'animation':[{type:7,name:'transformPlaceholder',definitions:[{type:0,name:'floating-ltr',
        styles:{type:6,styles:{top:'-22px',left:'-2px',transform:'scale(0.75)'},offset:(null as any)},
        options:(undefined as any)},{type:0,name:'floating-rtl',styles:{type:6,styles:{top:'-22px',
        left:'2px',transform:'scale(0.75)'},offset:(null as any)},options:(undefined as any)},
        {type:1,expr:'* => *',animation:{type:4,styles:(null as any),timings:'400ms cubic-bezier(0.25, 0.8, 0.25, 1)'},
            options:(null as any)}],options:{}},{type:7,name:'transformPanel',definitions:[{type:0,
        name:'showing',styles:{type:6,styles:{opacity:1,minWidth:'calc(100% + 32px)',
            transform:'scaleY(1)'},offset:(null as any)},options:(undefined as any)},
        {type:0,name:'showing-multiple',styles:{type:6,styles:{opacity:1,minWidth:'calc(100% + 64px)',
            transform:'scaleY(1)'},offset:(null as any)},options:(undefined as any)},
        {type:1,expr:'void => *',animation:[{type:6,styles:{opacity:0,minWidth:'100%',
            transform:'scaleY(0)'},offset:(null as any)},{type:4,styles:(null as any),
            timings:'150ms cubic-bezier(0.25, 0.8, 0.25, 1)'}],options:(null as any)},
        {type:1,expr:'* => void',animation:[{type:4,styles:{type:6,styles:{opacity:0},
            offset:(null as any)},timings:'250ms 100ms linear'}],options:(null as any)}],
        options:{}},{type:7,name:'fadeInContent',definitions:[{type:0,name:'showing',
        styles:{type:6,styles:{opacity:1},offset:(null as any)},options:(undefined as any)},
        {type:1,expr:'void => showing',animation:[{type:6,styles:{opacity:0},offset:(null as any)},
            {type:4,styles:(null as any),timings:'150ms 100ms cubic-bezier(0.55, 0, 0.55, 0.2)'}],
            options:(null as any)}],options:{}}]}});
function View_MdSelect_2(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'span',([] as any[]),
      (null as any),(null as any),(null as any),(null as any),(null as any))),(_l()(),
      i0.ɵted((null as any),['','']))],(null as any),(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.triggerValue;
    _ck(_v,1,0,currVal_0);
  });
}
function View_MdSelect_3(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[i0.ɵncd((null as any),0),(_l()(),i0.ɵand(0,(null as any),(null as any),
      0))],(null as any),(null as any));
}
function View_MdSelect_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),6,'span',[['class',
      'mat-select-value']],(null as any),(null as any),(null as any),(null as any),
      (null as any))),(_l()(),i0.ɵeld(0,(null as any),(null as any),5,'span',[['class',
      'mat-select-value-text']],(null as any),(null as any),(null as any),(null as any),
      (null as any))),i0.ɵdid(16384,(null as any),0,i2.NgSwitch,([] as any[]),{ngSwitch:[0,
      'ngSwitch']},(null as any)),(_l()(),i0.ɵand(16777216,(null as any),(null as any),
      1,(null as any),View_MdSelect_2)),i0.ɵdid(16384,(null as any),0,i2.NgSwitchDefault,
      [i0.ViewContainerRef,i0.TemplateRef,i2.NgSwitch],(null as any),(null as any)),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_MdSelect_3)),
      i0.ɵdid(278528,(null as any),0,i2.NgSwitchCase,[i0.ViewContainerRef,i0.TemplateRef,
          i2.NgSwitch],{ngSwitchCase:[0,'ngSwitchCase']},(null as any))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:boolean = !!_co.customTrigger;
    _ck(_v,2,0,currVal_0);
    const currVal_1:any = true;
    _ck(_v,6,0,currVal_1);
  },(null as any));
}
function View_MdSelect_4(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),3,'div',([] as any[]),
      [[24,'@transformPanel',0],[4,'transformOrigin',(null as any)],[2,'mat-select-panel-done-animating',
          (null as any)]],[[(null as any),'@transformPanel.done'],[(null as any),'keydown']],
      (_v,en,$event) => {
        var ad:boolean = true;
        var _co:any = _v.component;
        if (('@transformPanel.done' === en)) {
          const pd_0:any = ((<any>_co._onPanelDone()) !== false);
          ad = (pd_0 && ad);
        }
        if (('keydown' === en)) {
          const pd_1:any = ((<any>_co._handlePanelKeydown($event)) !== false);
          ad = (pd_1 && ad);
        }
        return ad;
      },(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i2.NgClass,[i0.IterableDiffers,
      i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{klass:[0,'klass'],ngClass:[1,
      'ngClass']},(null as any)),(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'div',
      [['class','mat-select-content']],[[24,'@fadeInContent',0]],[[(null as any),'@fadeInContent.done']],
      (_v,en,$event) => {
        var ad:boolean = true;
        var _co:any = _v.component;
        if (('@fadeInContent.done' === en)) {
          const pd_0:any = ((<any>_co._onFadeInDone()) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },(null as any),(null as any))),i0.ɵncd((null as any),1)],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_3:any = i0.ɵinlineInterpolate(1,'mat-select-panel ',('mat-' + _co.color),
        '');
    const currVal_4:any = _co.panelClass;
    _ck(_v,1,0,currVal_3,currVal_4);
  },(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = (_co.multiple? 'showing-multiple': 'showing');
    const currVal_1:any = _co._transformOrigin;
    const currVal_2:any = _co._panelDoneAnimating;
    _ck(_v,0,0,currVal_0,currVal_1,currVal_2);
    const currVal_5:any = 'showing';
    _ck(_v,2,0,currVal_5);
  });
}
export function View_MdSelect_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(2,[i0.ɵqud(402653184,1,{trigger:0}),i0.ɵqud(402653184,2,{overlayDir:0}),
      (_l()(),i0.ɵeld(0,[[1,0],['trigger',1]],(null as any),7,'div',[['aria-hidden',
          'true'],['cdk-overlay-origin',''],['class','mat-select-trigger']],(null as any),
          [[(null as any),'click']],(_v,en,$event) => {
            var ad:boolean = true;
            var _co:i1.MdSelect = _v.component;
            if (('click' === en)) {
              const pd_0:any = ((<any>_co.toggle()) !== false);
              ad = (pd_0 && ad);
            }
            return ad;
          },(null as any),(null as any))),i0.ɵdid(16384,[['origin',4]],0,i5.OverlayOrigin,
          [i0.ElementRef],(null as any),(null as any)),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),1,'span',[['class','mat-select-placeholder']],[[2,'mat-floating-placeholder',
              (null as any)],[24,'@transformPlaceholder',0],[4,'opacity',(null as any)],
              [4,'width','px']],(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['',''])),(_l()(),i0.ɵand(16777216,(null as any),
          (null as any),1,(null as any),View_MdSelect_1)),i0.ɵdid(16384,(null as any),
          0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),0,'span',[['class','mat-select-arrow']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),0,'span',[['class','mat-select-underline']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(_v,en,$event) => {
        var ad:boolean = true;
        var _co:i1.MdSelect = _v.component;
        if (('backdropClick' === en)) {
          const pd_0:any = ((<any>_co.close()) !== false);
          ad = (pd_0 && ad);
        }
        if (('attach' === en)) {
          const pd_1:any = ((<any>_co._onAttached()) !== false);
          ad = (pd_1 && ad);
        }
        if (('detach' === en)) {
          const pd_2:any = ((<any>_co.close()) !== false);
          ad = (pd_2 && ad);
        }
        return ad;
      },View_MdSelect_4)),i0.ɵdid(671744,[[2,4]],0,i5.ConnectedOverlayDirective,[i5.Overlay,
          i0.Renderer2,i0.TemplateRef,i0.ViewContainerRef,i5.ɵc,[2,i6.Directionality]],
          {_deprecatedOrigin:[0,'_deprecatedOrigin'],_deprecatedPositions:[1,'_deprecatedPositions'],
              _deprecatedOffsetY:[2,'_deprecatedOffsetY'],_deprecatedMinWidth:[3,'_deprecatedMinWidth'],
              _deprecatedBackdropClass:[4,'_deprecatedBackdropClass'],_deprecatedScrollStrategy:[5,
                  '_deprecatedScrollStrategy'],_deprecatedOpen:[6,'_deprecatedOpen'],
              _deprecatedHasBackdrop:[7,'_deprecatedHasBackdrop']},{backdropClick:'backdropClick',
              attach:'attach',detach:'detach'})],(_ck,_v) => {
    var _co:i1.MdSelect = _v.component;
    const currVal_5:any = _co._hasValue();
    _ck(_v,7,0,currVal_5);
    const currVal_6:any = i0.ɵnov(_v,3);
    const currVal_7:any = _co._positions;
    const currVal_8:any = _co._offsetY;
    const currVal_9:any = _co._triggerWidth;
    const currVal_10:any = 'cdk-overlay-transparent-backdrop';
    const currVal_11:any = _co._scrollStrategy;
    const currVal_12:any = _co.panelOpen;
    const currVal_13:any = '';
    _ck(_v,11,0,currVal_6,currVal_7,currVal_8,currVal_9,currVal_10,currVal_11,currVal_12,
        currVal_13);
  },(_ck,_v) => {
    var _co:i1.MdSelect = _v.component;
    const currVal_0:any = _co._hasValue();
    const currVal_1:any = _co._getPlaceholderAnimationState();
    const currVal_2:any = _co._getPlaceholderOpacity();
    const currVal_3:any = _co._selectedValueWidth;
    _ck(_v,4,0,currVal_0,currVal_1,currVal_2,currVal_3);
    const currVal_4:any = _co.placeholder;
    _ck(_v,5,0,currVal_4);
  });
}
export function View_MdSelect_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),4,'md-select',[['class',
      'mat-select'],['role','listbox']],[[1,'tabindex',0],[1,'aria-label',0],[1,'aria-labelledby',
      0],[1,'aria-required',0],[1,'aria-disabled',0],[1,'aria-invalid',0],[1,'aria-owns',
      0],[1,'aria-multiselectable',0],[2,'mat-select-disabled',(null as any)],[2,'mat-select-invalid',
      (null as any)],[2,'mat-select-required',(null as any)]],[[(null as any),'keydown'],
      [(null as any),'blur']],(_v,en,$event) => {
    var ad:boolean = true;
    if (('keydown' === en)) {
      const pd_0:any = ((<any>i0.ɵnov(_v,1)._handleClosedKeydown($event)) !== false);
      ad = (pd_0 && ad);
    }
    if (('blur' === en)) {
      const pd_1:any = ((<any>i0.ɵnov(_v,1)._onBlur()) !== false);
      ad = (pd_1 && ad);
    }
    return ad;
  },View_MdSelect_0,RenderType_MdSelect)),i0.ɵdid(1294336,(null as any),3,i1.MdSelect,
      [i4.ViewportRuler,i0.ChangeDetectorRef,i3.Platform,i0.Renderer2,i0.ElementRef,
          [2,i6.Directionality],[2,i10.NgForm],[2,i10.FormGroupDirective],[8,(null as any)],
          [8,(null as any)],[2,i9.MD_PLACEHOLDER_GLOBAL_OPTIONS],i1.MD_SELECT_SCROLL_STRATEGY],
      (null as any),(null as any)),i0.ɵqud(603979776,1,{options:1}),i0.ɵqud(603979776,
      2,{optionGroups:1}),i0.ɵqud(335544320,3,{customTrigger:0})],(_ck,_v) => {
    _ck(_v,1,0);
  },(_ck,_v) => {
    const currVal_0:any = i0.ɵnov(_v,1).tabIndex;
    const currVal_1:any = i0.ɵnov(_v,1)._ariaLabel;
    const currVal_2:any = i0.ɵnov(_v,1).ariaLabelledby;
    const currVal_3:any = i0.ɵnov(_v,1).required.toString();
    const currVal_4:any = i0.ɵnov(_v,1).disabled.toString();
    const currVal_5:any = i0.ɵnov(_v,1)._isErrorState();
    const currVal_6:any = i0.ɵnov(_v,1)._optionIds;
    const currVal_7:any = i0.ɵnov(_v,1).multiple;
    const currVal_8:any = i0.ɵnov(_v,1).disabled;
    const currVal_9:any = i0.ɵnov(_v,1)._isErrorState();
    const currVal_10:any = i0.ɵnov(_v,1).required;
    _ck(_v,0,1,[currVal_0,currVal_1,currVal_2,currVal_3,currVal_4,currVal_5,currVal_6,
        currVal_7,currVal_8,currVal_9,currVal_10]);
  });
}
export const MdSelectNgFactory:i0.ComponentFactory<i1.MdSelect> = i0.ɵccf('md-select, mat-select',
    i1.MdSelect,View_MdSelect_Host_0,{color:'color',disabled:'disabled',tabIndex:'tabIndex',
        panelClass:'panelClass',placeholder:'placeholder',required:'required',multiple:'multiple',
        compareWith:'compareWith',floatPlaceholder:'floatPlaceholder',value:'value',
        disableRipple:'disableRipple',ariaLabel:'aria-label',ariaLabelledby:'aria-labelledby'},
    {onOpen:'onOpen',onClose:'onClose',change:'change',valueChange:'valueChange'},
    ['md-select-trigger, mat-select-trigger','*']);
