// @angular/material/snack-bar/typings/index.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from '@angular/material/snack-bar';
import * as i2 from '@angular/cdk/platform';
import * as i3 from '@angular/cdk/scrolling';
import * as i4 from '@angular/cdk/overlay';
import * as i5 from '@angular/common';
import * as i6 from '@angular/cdk/bidi';
import * as i7 from '@angular/platform-browser';
import * as i8 from '@angular/cdk/a11y';
import * as i9 from '@angular/cdk/portal';
import * as i10 from '@angular/material/core';
export const MdSnackBarModuleNgFactory:i0.NgModuleFactory<i1.MdSnackBarModule> = i0.ɵcmf(i1.MdSnackBarModule,
    ([] as any[]),(_l:any) => {
      return i0.ɵmod([i0.ɵmpd(512,i0.ComponentFactoryResolver,i0.ɵCodegenComponentFactoryResolver,
          [[8,[MdSnackBarContainerNgFactory,SimpleSnackBarNgFactory]],[3,i0.ComponentFactoryResolver],
              i0.NgModuleRef]),i0.ɵmpd(4608,i2.Platform,i2.Platform,([] as any[])),
          i0.ɵmpd(5120,i3.ScrollDispatcher,i3.SCROLL_DISPATCHER_PROVIDER_FACTORY,[[3,
              i3.ScrollDispatcher],i0.NgZone,i2.Platform]),i0.ɵmpd(5120,i3.ViewportRuler,
              i3.VIEWPORT_RULER_PROVIDER_FACTORY,[[3,i3.ViewportRuler],i3.ScrollDispatcher]),
          i0.ɵmpd(4608,i4.ScrollStrategyOptions,i4.ScrollStrategyOptions,[i3.ScrollDispatcher,
              i3.ViewportRuler]),i0.ɵmpd(5120,i4.OverlayContainer,i4.ɵa,[[3,i4.OverlayContainer]]),
          i0.ɵmpd(4608,i4.ɵf,i4.ɵf,[i3.ViewportRuler]),i0.ɵmpd(4608,i4.Overlay,i4.Overlay,
              [i4.ScrollStrategyOptions,i4.OverlayContainer,i0.ComponentFactoryResolver,
                  i4.ɵf,i0.ApplicationRef,i0.Injector,i0.NgZone]),i0.ɵmpd(5120,i4.ɵc,
              i4.ɵd,[i4.Overlay]),i0.ɵmpd(4608,i5.NgLocalization,i5.NgLocaleLocalization,
              [i0.LOCALE_ID]),i0.ɵmpd(6144,i6.DIR_DOCUMENT,(null as any),[i7.DOCUMENT]),
          i0.ɵmpd(4608,i6.Directionality,i6.Directionality,[[2,i6.DIR_DOCUMENT]]),
          i0.ɵmpd(5120,i8.LiveAnnouncer,i8.LIVE_ANNOUNCER_PROVIDER_FACTORY,[[3,i8.LiveAnnouncer],
              [2,i8.LIVE_ANNOUNCER_ELEMENT_TOKEN],i2.Platform]),i0.ɵmpd(4608,i1.MdSnackBar,
              i1.MdSnackBar,[i4.Overlay,i8.LiveAnnouncer,i0.Injector,[3,i1.MdSnackBar]]),
          i0.ɵmpd(512,i9.PortalModule,i9.PortalModule,([] as any[])),i0.ɵmpd(512,i2.PlatformModule,
              i2.PlatformModule,([] as any[])),i0.ɵmpd(512,i3.ScrollDispatchModule,
              i3.ScrollDispatchModule,([] as any[])),i0.ɵmpd(512,i4.OverlayModule,
              i4.OverlayModule,([] as any[])),i0.ɵmpd(512,i5.CommonModule,i5.CommonModule,
              ([] as any[])),i0.ɵmpd(512,i10.CompatibilityModule,i10.CompatibilityModule,
              ([] as any[])),i0.ɵmpd(512,i6.BidiModule,i6.BidiModule,([] as any[])),
          i0.ɵmpd(256,i10.MATERIAL_SANITY_CHECKS,true,([] as any[])),i0.ɵmpd(512,i10.MdCommonModule,
              i10.MdCommonModule,[[2,i7.DOCUMENT],[2,i10.MATERIAL_SANITY_CHECKS]]),
          i0.ɵmpd(512,i1.MdSnackBarModule,i1.MdSnackBarModule,([] as any[]))]);
    });
const styles_MdSnackBarContainer:any[] = ['.mat-snack-bar-container{border-radius:2px;box-sizing:content-box;display:block;margin:24px;max-width:568px;min-width:288px;padding:14px 24px;transform:translateY(100%)}.mat-snack-bar-container.mat-snack-bar-center{margin:0}.mat-snack-bar-container.mat-snack-bar-top{transform:translateY(-100%)}@media screen and (-ms-high-contrast:active){.mat-snack-bar-container{border:solid 1px}}'];
export const RenderType_MdSnackBarContainer:i0.RendererType2 = i0.ɵcrt({encapsulation:2,
    styles:styles_MdSnackBarContainer,data:{'animation':[{type:7,name:'state',definitions:[{type:0,
        name:'visible-top',styles:{type:6,styles:{transform:'translateY(0%)'},offset:(null as any)},
        options:(undefined as any)},{type:0,name:'hidden-top',styles:{type:6,styles:{transform:'translateY(-100%)'},
        offset:(null as any)},options:(undefined as any)},{type:1,expr:'visible-top => hidden-top',
        animation:{type:4,styles:(null as any),timings:'195ms cubic-bezier(0.0,0.0,0.2,1)'},
        options:(null as any)},{type:1,expr:'void => visible-top',animation:{type:4,
        styles:(null as any),timings:'225ms cubic-bezier(0.4,0.0,1,1)'},options:(null as any)},
        {type:0,name:'visible-bottom',styles:{type:6,styles:{transform:'translateY(0%)'},
            offset:(null as any)},options:(undefined as any)},{type:0,name:'hidden-bottom',
            styles:{type:6,styles:{transform:'translateY(100%)'},offset:(null as any)},
            options:(undefined as any)},{type:1,expr:'visible-bottom => hidden-bottom',
            animation:{type:4,styles:(null as any),timings:'195ms cubic-bezier(0.0,0.0,0.2,1)'},
            options:(null as any)},{type:1,expr:'void => visible-bottom',animation:{type:4,
            styles:(null as any),timings:'225ms cubic-bezier(0.4,0.0,1,1)'},options:(null as any)}],
        options:{}}]}});
function View_MdSnackBarContainer_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵand(0,(null as any),(null as any),0))],(null as any),
      (null as any));
}
export function View_MdSnackBarContainer_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(2,[i0.ɵqud(402653184,1,{_portalHost:0}),(_l()(),i0.ɵand(16777216,
      (null as any),(null as any),1,(null as any),View_MdSnackBarContainer_1)),i0.ɵdid(147456,
      [[1,4]],0,i9.PortalHostDirective,[i0.ComponentFactoryResolver,i0.ViewContainerRef],
      {portal:[0,'portal']},(null as any))],(_ck,_v) => {
    const currVal_0:any = '';
    _ck(_v,2,0,currVal_0);
  },(null as any));
}
export function View_MdSnackBarContainer_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'snack-bar-container',
      [['class','mat-snack-bar-container'],['role','alert']],[[40,'@state',0]],[['component',
          '@state.done']],(_v,en,$event) => {
        var ad:boolean = true;
        if (('component:@state.done' === en)) {
          const pd_0:any = ((<any>i0.ɵnov(_v,1).onAnimationEnd($event)) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },View_MdSnackBarContainer_0,RenderType_MdSnackBarContainer)),i0.ɵdid(180224,
      (null as any),0,i1.MdSnackBarContainer,[i0.NgZone,i0.Renderer2,i0.ElementRef,
          i0.ChangeDetectorRef],(null as any),(null as any))],(null as any),(_ck,_v) => {
    const currVal_0:any = i0.ɵnov(_v,1).getAnimationState();
    _ck(_v,0,0,currVal_0);
  });
}
export const MdSnackBarContainerNgFactory:i0.ComponentFactory<i1.MdSnackBarContainer> = i0.ɵccf('snack-bar-container',
    i1.MdSnackBarContainer,View_MdSnackBarContainer_Host_0,{},{},([] as any[]));
const styles_SimpleSnackBar:any[] = ['.mat-simple-snackbar{display:flex;justify-content:space-between;line-height:20px}.mat-simple-snackbar-action{-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;cursor:pointer;outline:0;border:none;-webkit-tap-highlight-color:transparent;background:0 0;flex-shrink:0;margin-left:48px}[dir=rtl] .mat-simple-snackbar-action{margin-right:48px;margin-left:0}'];
export const RenderType_SimpleSnackBar:i0.RendererType2 = i0.ɵcrt({encapsulation:2,
    styles:styles_SimpleSnackBar,data:{}});
function View_SimpleSnackBar_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'button',[['class',
      'mat-simple-snackbar-action']],(null as any),[[(null as any),'click']],(_v,en,
      $event) => {
    var ad:boolean = true;
    var _co:any = _v.component;
    if (('click' === en)) {
      const pd_0:any = ((<any>_co.action()) !== false);
      ad = (pd_0 && ad);
    }
    return ad;
  },(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['','']))],(null as any),
      (_ck,_v) => {
        var _co:any = _v.component;
        const currVal_0:any = _co.data.action;
        _ck(_v,1,0,currVal_0);
      });
}
export function View_SimpleSnackBar_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(2,[(_l()(),i0.ɵted((null as any),['',' '])),(_l()(),i0.ɵand(16777216,
      (null as any),(null as any),1,(null as any),View_SimpleSnackBar_1)),i0.ɵdid(16384,
      (null as any),0,i5.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},
      (null as any))],(_ck,_v) => {
    var _co:i1.SimpleSnackBar = _v.component;
    const currVal_1:any = _co.hasAction;
    _ck(_v,2,0,currVal_1);
  },(_ck,_v) => {
    var _co:i1.SimpleSnackBar = _v.component;
    const currVal_0:any = _co.data.message;
    _ck(_v,0,0,currVal_0);
  });
}
export function View_SimpleSnackBar_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'simple-snack-bar',
      [['class','mat-simple-snackbar']],(null as any),(null as any),(null as any),
      View_SimpleSnackBar_0,RenderType_SimpleSnackBar)),i0.ɵdid(49152,(null as any),
      0,i1.SimpleSnackBar,[i1.MdSnackBarRef,i1.MD_SNACK_BAR_DATA],(null as any),(null as any))],
      (null as any),(null as any));
}
export const SimpleSnackBarNgFactory:i0.ComponentFactory<i1.SimpleSnackBar> = i0.ɵccf('simple-snack-bar',
    i1.SimpleSnackBar,View_SimpleSnackBar_Host_0,{},{},([] as any[]));
